{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Importing tools \n",
    "SQL commands are used throughout this lab in order to create, select, drop or insert data into the tables made (Tables: GENES, ENZYMES, PATHWAYS). Sqlite3 is imported in order to introduce the SQL commands. \"Database\" is made referring to the database made to contain the tables. The cursor is for reading and performing actions for the database made. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sqlite3\n",
    "conn= sqlite3.connect('Database.db')\n",
    "c = conn.cursor()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Search for data imported from Entrez"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Glycolysis enzymes:\n",
      "Phosphoglucomutase (EC 5.4.2.2) [Klebsiella aerogenes EA1509E]\n",
      "phosphoglucomutase (ec 5.4.2.2) [Photorhabdus asymbiotica subsp. asymbiotica ATCC 43949]\n",
      "Phosphoglucomutase (EC 5.4.2.2) [Enterococcus faecium]\n",
      "Phosphoglucomutase (EC 5.4.2.2) [Streptococcus pneumoniae SPNA45]\n",
      "phosphoglucomutase [Enterobacter roggenkampii EC_38VIM1]\n",
      "phosphoglucomutase [Escherichia coli]\n",
      "phosphoglucomutase [Escherichia coli]\n",
      "phosphoglucomutase [Escherichia coli]\n",
      "alpha-phosphoglucomutase / phosphomannomutase [Methanocella conradii HZ254]\n",
      "phophoglucomutase-1 protein, putative [Cryptosporidium muris RN66]\n",
      "pgm [Corynebacterium jeikeium K411]\n",
      "Phosphoglucomutase [Bartonella tribocorum CIP 105476]\n",
      "phosphoglucomutase [Neisseria meningitidis Z2491]\n",
      "phosphoglucomutase [Ralstonia pickettii DTP0602]\n",
      "phosphoglucomutase, alpha-D-glucose phosphate-specific [Vibrio cholerae O1 str. EC-0051]\n",
      "phosphoglucomutase, alpha-D-glucose phosphate-specific [Vibrio cholerae O1 str. EC-0012]\n",
      "phosphoglucomutase, alpha-D-glucose phosphate-specific [Vibrio cholerae O1 str. EC-0027]\n",
      "phosphoglucomutase, alpha-D-glucose phosphate-specific [Vibrio cholerae O1 str. EC-0009]\n",
      "phophoglucomutase-1 protein [Cryptosporidium muris RN66]\n",
      "alpha-D-glucose phosphate-specific phosphoglucomutase [Enterobacter sp. EC-NT1]\n",
      "\n",
      "\n",
      "Aldose 1-epimerase (EC 5.1.3.3) [Klebsiella aerogenes EA1509E]\n",
      "Aldose 1-epimerase (EC 5.1.3.3) [Enterococcus faecium]\n",
      "Aldose 1-epimerase (EC 5.1.3.3) [Enterococcus faecium]\n",
      "Aldose 1-epimerase (EC 5.1.3.3) [Enterococcus faecium]\n",
      "galactose-1-epimerase [Enterobacter roggenkampii EC_38VIM1]\n",
      "aldose 1-epimerase [Bradyrhizobium diazoefficiens USDA 110]\n",
      "aldose 1-epimerase [Rhodopirellula baltica SH 1]\n",
      "aldose 1-epimerase [Rhodopirellula baltica SH 1]\n",
      "Aldose 1-epimerase [Escherichia coli]\n",
      "Aldose 1-epimerase [Escherichia coli]\n",
      "Aldose 1-epimerase [Escherichia coli]\n",
      "putative aldose 1-epimerase [Staphylococcus carnosus subsp. carnosus TM300]\n",
      "aldose 1-epimerase [Rhodopirellula baltica SH 1]\n",
      "probable aldose 1-epimerase [Rhodopirellula baltica SH 1]\n",
      "Aldose 1-epimerase [Lactobacillus salivarius cp400]\n",
      "aldose 1-epimerase [Vibrio cholerae O1 str. EC-0051]\n",
      "aldose 1-epimerase [Vibrio cholerae O1 str. EC-0027]\n",
      "aldose 1-epimerase [Vibrio cholerae O1 str. EC-0009]\n",
      "Aldose 1-epimerase [Escherichia coli]\n",
      "Aldose 1-epimerase [Escherichia coli]\n",
      "\n",
      "\n",
      "Enolase (EC 4.2.1.11) [Paraburkholderia rhizoxinica HKI 454]\n",
      "Enolase (EC 4.2.1.11) [Candidatus Synechococcus spongiarum]\n",
      "Enolase (EC 4.2.1.11) [Kibdelosporangium sp. MJ126-NF4]\n",
      "Enolase (EC 4.2.1.11) [Kibdelosporangium sp. MJ126-NF4]\n",
      "Enolase (EC 4.2.1.11) [Klebsiella aerogenes EA1509E]\n",
      "phosphopyruvate hydratase (EC 4.2.1.11) [imported] - Buchnera sp. (strain APS)\n",
      "phosphopyruvate hydratase (EC 4.2.1.11) - yeast (Candida albicans) (fragment)\n",
      "phosphopyruvate hydratase (EC 4.2.1.11) - rat\n",
      "phosphopyruvate hydratase (EC 4.2.1.11) alpha / tau-crystallin - slider turtle (fragments)\n",
      "phosphopyruvate hydratase (EC 4.2.1.11) alpha / tau-crystallin - sea lamprey (fragments)\n",
      "Enolase (EC 4.2.1.11) [Enterococcus faecium]\n",
      "Enolase (EC 4.2.1.11) (plasmid) [Pseudomonas fluorescens SBW25]\n",
      "eno enolase (2-phosphoglycerate dehydratase) (2-phospho-D-glycerate hydro-lyase) (EC 4.2.1.11) [Pyrococcus abyssi GE5]\n",
      "muscle-specific enolase beta subunit (EC 4.2.1.11), partial [Mus musculus]\n",
      "tau-crystallin/alpha-enolase (EC 4.2.1.11) [Anas platyrhynchos]\n",
      "alpha enolase (EC 4.2.1.11) [Homo sapiens]\n",
      "enolase, 2-phospho-D-glycerate hydrolase {N-terminal} {EC 4.2.1.11} [Thermotoga maritima, Peptide Partial, 24 aa]\n",
      "enolase, 2-phospho-D-glycerate hydrolyase {EC 4.2.1.11} [Pyrococcus furiosus, Peptide Partial, 27 aa]\n",
      "enolase {internal fragment} {EC 4.2.1.11} [Drosophila melanogaster, CME W2 wing imaginal disc cell line, Peptide Partial, 17 aa]\n",
      "immunoglobulin production stimulating factor-II beta, IPSF-II beta, alpha-enolase {internal fragment} {EC 4.2.1.11} [human, Burkitt's lymphoma Namalwa cells, Peptide Partial, 26 aa]\n",
      "\n",
      "\n",
      "Glucose-6-phosphate isomerase (EC 5.3.1.9) / Glucose-6 phosphate 1-epimerase (EC 5.1.3.15) [Paraburkholderia rhizoxinica HKI 454]\n",
      "Glucose-6-phosphate isomerase (EC 5.3.1.9) [Candidatus Synechococcus spongiarum]\n",
      "Glucose-6-phosphate isomerase (EC 5.3.1.9) [Kibdelosporangium sp. MJ126-NF4]\n",
      "Glucose-6-phosphate isomerase (EC 5.3.1.9) [Klebsiella aerogenes EA1509E]\n",
      "Glucose-6-phosphate isomerase (EC 5.3.1.9) (GPI) (Phosphoglucose isomerase) (PGI) (Phosphohexose isomerase) (PHI) [Propionibacterium freudenreichii subsp. shermanii CIRM-BIA1]\n"
     ]
    },
    {
     "ename": "HTTPError",
     "evalue": "HTTP Error 400: Bad Request",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mHTTPError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-159-b5a256142815>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     49\u001b[0m                        idtype = 'acc')\n\u001b[1;32m     50\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mEntrez\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'IdList'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 51\u001b[0;31m     \u001b[0mhandle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mEntrez\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mefetch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'protein'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mid\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrettype\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'gb'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mretmode\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'text'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     52\u001b[0m     \u001b[0mtemp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mSeqIO\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'gb'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     53\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtemp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdescription\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/anaconda3-5.2.0/lib/python3.6/site-packages/Bio/Entrez/__init__.py\u001b[0m in \u001b[0;36mefetch\u001b[0;34m(db, **keywords)\u001b[0m\n\u001b[1;32m    182\u001b[0m             \u001b[0;31m# more than about 200 IDs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    183\u001b[0m             \u001b[0mpost\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 184\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_open\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcgi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvariables\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpost\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpost\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    185\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/anaconda3-5.2.0/lib/python3.6/site-packages/Bio/Entrez/__init__.py\u001b[0m in \u001b[0;36m_open\u001b[0;34m(cgi, params, post, ecitmatch)\u001b[0m\n\u001b[1;32m    531\u001b[0m             \u001b[0mhandle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_urlopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcgi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    532\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0m_HTTPError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mexception\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 533\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mexception\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    534\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    535\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0m_binary_to_string_handle\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/anaconda3-5.2.0/lib/python3.6/site-packages/Bio/Entrez/__init__.py\u001b[0m in \u001b[0;36m_open\u001b[0;34m(cgi, params, post, ecitmatch)\u001b[0m\n\u001b[1;32m    529\u001b[0m             \u001b[0mhandle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_urlopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcgi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0m_as_bytes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    530\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 531\u001b[0;31m             \u001b[0mhandle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_urlopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcgi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    532\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0m_HTTPError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mexception\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    533\u001b[0m         \u001b[0;32mraise\u001b[0m \u001b[0mexception\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/anaconda3-5.2.0/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(url, data, timeout, cafile, capath, cadefault, context)\u001b[0m\n\u001b[1;32m    221\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    222\u001b[0m         \u001b[0mopener\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_opener\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 223\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mopener\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    224\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    225\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0minstall_opener\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mopener\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/anaconda3-5.2.0/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36mopen\u001b[0;34m(self, fullurl, data, timeout)\u001b[0m\n\u001b[1;32m    530\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mprocessor\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprocess_response\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprotocol\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    531\u001b[0m             \u001b[0mmeth\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprocessor\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 532\u001b[0;31m             \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmeth\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    533\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    534\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/anaconda3-5.2.0/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36mhttp_response\u001b[0;34m(self, request, response)\u001b[0m\n\u001b[1;32m    640\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0;36m200\u001b[0m \u001b[0;34m<=\u001b[0m \u001b[0mcode\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m300\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    641\u001b[0m             response = self.parent.error(\n\u001b[0;32m--> 642\u001b[0;31m                 'http', request, response, code, msg, hdrs)\n\u001b[0m\u001b[1;32m    643\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    644\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresponse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/anaconda3-5.2.0/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36merror\u001b[0;34m(self, proto, *args)\u001b[0m\n\u001b[1;32m    568\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mhttp_err\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    569\u001b[0m             \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mdict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'default'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'http_error_default'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0morig_args\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 570\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_call_chain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    571\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    572\u001b[0m \u001b[0;31m# XXX probably also want an abstract factory that knows when it makes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/anaconda3-5.2.0/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36m_call_chain\u001b[0;34m(self, chain, kind, meth_name, *args)\u001b[0m\n\u001b[1;32m    502\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mhandler\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mhandlers\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    503\u001b[0m             \u001b[0mfunc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandler\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 504\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    505\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    506\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/anaconda3-5.2.0/lib/python3.6/urllib/request.py\u001b[0m in \u001b[0;36mhttp_error_default\u001b[0;34m(self, req, fp, code, msg, hdrs)\u001b[0m\n\u001b[1;32m    648\u001b[0m \u001b[0;32mclass\u001b[0m \u001b[0mHTTPDefaultErrorHandler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mBaseHandler\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    649\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mhttp_error_default\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmsg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhdrs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 650\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mHTTPError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfull_url\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmsg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhdrs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    651\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    652\u001b[0m \u001b[0;32mclass\u001b[0m \u001b[0mHTTPRedirectHandler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mBaseHandler\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mHTTPError\u001b[0m: HTTP Error 400: Bad Request"
     ]
    }
   ],
   "source": [
    "# Finding the enzymes for each pathway by examining the KEGG Glycolysis / Gluconeogenesis - Reference pathway\n",
    "# picked out several chosen enzyme commission numbers indicated for each pathway.\n",
    "# use Entrez search to find the exact name for each enzyme commission number and the corresponding included organisms. \n",
    "# Google for each enzyme's function. \n",
    "glycolysis_terms = ['EC 5.4.2.2', 'EC 5.1.3.3', 'EC 4.2.1.11' 'EC 5.3.1.9']\n",
    "pentose_phosphate_terms = ['EC 5.3.1.9', 'EC 2.7.1.11', 'EC 4.1.2.13', 'EC 3.1.3.11']\n",
    "citric_acid_terms = ['EC 2.3.1.12', 'EC 1.2.4.1', 'EC 1.8.1.4', '4.1.1.49', '4.1.1.32']\n",
    "\n",
    "print(\"Glycolysis enzymes:\")\n",
    "from Bio import SeqIO\n",
    "from Bio import Entrez\n",
    "Entrez.email = 'joanne91218@berkeley.edu'\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'EC 5.4.2.2',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)    \n",
    "print('\\n')\n",
    "\n",
    "\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'EC 5.1.3.3',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')    \n",
    "\n",
    "\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'EC 4.2.1.11',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')\n",
    "\n",
    "\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'EC 5.3.1.9',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')\n",
    "\n",
    "print('Pentose Phosphate Pathway enzymes:')\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'EC 5.3.1.9',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')\n",
    "\n",
    "\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'EC 2.7.1.11',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')\n",
    "\n",
    "\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'EC 4.1.2.13',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')\n",
    "\n",
    "\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'EC 3.1.3.11',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')\n",
    "\n",
    "print('Citric acid cycle enzymes:')\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'EC 2.3.1.12',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')\n",
    "\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'EC 1.2.4.1',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')\n",
    "\n",
    "\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'EC 1.8.1.4',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')\n",
    "\n",
    "\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = '4.1.1.49',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')\n",
    "\n",
    "\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = '4.1.1.32',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp.description)\n",
    "print('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: M83088.1\n",
      "Name: HUMPGM1A\n",
      "Description: Human phosphoglucomutase 1 (PGM1) mRNA, complete cds\n",
      "Number of features: 3\n",
      "/molecule_type=mRNA\n",
      "/topology=linear\n",
      "/data_file_division=PRI\n",
      "/date=07-JAN-1995\n",
      "/accessions=['M83088']\n",
      "/sequence_version=1\n",
      "/keywords=['phosphoglucomutase']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='Phosphoglucomutase 1: complete human and rabbit mRNA sequences and direct mapping of this highly polymorphic marker on human chromosome 1', ...)]\n",
      "/comment=Original source text: Homo sapiens Adult Muscle, skeletal cDNA to\n",
      "mRNA.\n",
      "Seq('GGGCCGGCCGCCCCTCCGCCAGCCAAGTCCGCCGCTCTGACCCCCGGCAGCAAG...ACA', IUPACAmbiguousDNA())\n",
      "ID: S67989.1\n",
      "Name: S67989\n",
      "Description: phosphoglucomutase 1 {alternatively spliced, exon 1A, promoter} [human, Genomic, 900 nt]\n",
      "Number of features: 4\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=PRI\n",
      "/date=05-MAY-2000\n",
      "/accessions=['S67989']\n",
      "/sequence_version=1\n",
      "/keywords=['']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='Phosphoglucomutase 1: a gene with two promoters and a duplicated first exon', ...)]\n",
      "Seq('GCCTCTTTTTACAGTGGATAGGATTCCGAAGGGAGAGTAAATCAGTAAGGCTCC...GCA', IUPACAmbiguousDNA())\n",
      "ID: KJ905858.1\n",
      "Name: KJ905858\n",
      "Description: Synthetic construct Homo sapiens clone ccsbBroadEn_15528 PGM1 gene, encodes complete protein\n",
      "Number of features: 3\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=SYN\n",
      "/date=19-MAR-2015\n",
      "/accessions=['KJ905858']\n",
      "/sequence_version=1\n",
      "/keywords=['ORFeome Collaboration', 'Gateway cloning system', 'full-length ORF without stop codon']\n",
      "/source=synthetic construct\n",
      "/organism=synthetic construct\n",
      "/taxonomy=['other sequences', 'artificial sequences']\n",
      "/references=[Reference(title='A public genome-scale lentiviral expression library of human ORFs', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=This clone is distributed via members of the ORFeome Collaboration\n",
      "(http://www.orfeomecollaboration.org/), including the\n",
      "Dana-Farber/Harvard Cancer Center (DF/HCC) DNA Resource Core DNA\n",
      "Repository\n",
      "(http://www.dfhcc.harvard.edu/core-facilities/dnaresource/) and the\n",
      "DNASU Plasmid Repository at ASU Biodesign Institute\n",
      "(http://dnasu.asu.edu/DNASU/Home.jsp).  See also\n",
      "http://www.broadinstitute.org/rnai/public/ for additional\n",
      "information on this clone.\n",
      "Seq('GTTCGTTGCAACAAATTGATGAGCAATGCTTTTTTATAATGCCAACTTTGTACA...AAC', IUPACAmbiguousDNA())\n",
      "ID: S67998.1\n",
      "Name: S67998\n",
      "Description: phosphoglucomutase 1 {alternatively spliced, exon 1B, promoter} [human, Genomic, 1050 nt]\n",
      "Number of features: 4\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=PRI\n",
      "/date=26-JUL-2016\n",
      "/accessions=['S67998']\n",
      "/sequence_version=1\n",
      "/keywords=['']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='Phosphoglucomutase 1: a gene with two promoters and a duplicated first exon', ...)]\n",
      "Seq('TTGCATGTGTGAAGTAAAAATAATTTCTTAGGTTTCAAAATGACAGTGATGTAA...GGT', IUPACAmbiguousDNA())\n",
      "ID: NM_001172818.1\n",
      "Name: NM_001172818\n",
      "Description: Homo sapiens phosphoglucomutase 1 (PGM1), transcript variant 2, mRNA\n",
      "Number of features: 21\n",
      "/molecule_type=mRNA\n",
      "/topology=linear\n",
      "/data_file_division=PRI\n",
      "/date=24-JUN-2018\n",
      "/accessions=['NM_001172818']\n",
      "/sequence_version=1\n",
      "/keywords=['RefSeq']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='Asp263 missense variants perturb the active site of human phosphoglucomutase 1', ...), Reference(title='Induced Structural Disorder as a Molecular Mechanism for Enzyme Dysfunction in Phosphoglucomutase 1 Deficiency', ...), Reference(title='Phosphoglucomutase-1 deficiency: Intrafamilial clinical variability and common secondary adrenal insufficiency', ...), Reference(title='Compromised catalysis and potential folding defects in in vitro studies of missense mutants associated with hereditary phosphoglucomutase 1 deficiency', ...), Reference(title='Phosphoglucomutase1 is necessary for sustained cell growth under repetitive glucose depletion', ...), Reference(title='Congenital Disorders of N-Linked Glycosylation and Multiple Pathway Overview', ...), Reference(title='Treatment of Haemophilus aphrophilus endocarditis with ciprofloxacin', ...), Reference(title='Phosphoglucomutase 1: complete human and rabbit mRNA sequences and direct mapping of this highly polymorphic marker on human chromosome 1', ...), Reference(title='Antigenic analysis of the major human phosphoglucomutase isozymes: PGM1, PGM2, PGM3 and PGM4', ...), Reference(title='Gene localisation of the PGM1 enzyme system and the Duffy blood groups on chromosome No. 1 by means of a new fragile site at 1p31', ...)]\n",
      "/comment=REVIEWED REFSEQ: This record has been curated by NCBI staff. The\n",
      "reference sequence was derived from DC373587.1, AK300100.1 and\n",
      "BC019920.1.\n",
      "Summary: The protein encoded by this gene is an isozyme of\n",
      "phosphoglucomutase (PGM) and belongs to the phosphohexose mutase\n",
      "family. There are several PGM isozymes, which are encoded by\n",
      "different genes and catalyze the transfer of phosphate between the\n",
      "1 and 6 positions of glucose. In most cell types, this PGM isozyme\n",
      "is predominant, representing about 90% of total PGM activity. In\n",
      "red cells, PGM2 is a major isozyme. This gene is highly\n",
      "polymorphic. Mutations in this gene cause glycogen storage disease\n",
      "type 14. Alternativley spliced transcript variants encoding\n",
      "different isoforms have been identified in this gene.[provided by\n",
      "RefSeq, Mar 2010].\n",
      "Transcript Variant: This variant (2) has an alternate 5' exon, as\n",
      "compared to variant 1. The resulting isoform (2) has a longer and\n",
      "different N-terminus, as compared to isoform 1.\n",
      "Publication Note:  This RefSeq record includes a subset of the\n",
      "publications that are available for this gene. Please see the Gene\n",
      "record to access additional publications.\n",
      "                               [ECO:0000332]\n",
      "                               SAMEA2145313, SAMEA2145544\n",
      "                               [ECO:0000348]\n",
      "/structured_comment=OrderedDict([('Evidence-Data', OrderedDict([('Transcript exon combination', 'SRR1803611.99027.1, AK300100.1'), ('RNAseq introns', 'single sample supports all introns')]))])\n",
      "Seq('AAAACAACTTTGAAGACCTGCTTTGTGTGTGGGTGCGAGTGGGTGTGTGGCCCT...AAA', IUPACAmbiguousDNA())\n",
      "ID: NM_002633.2\n",
      "Name: NM_002633\n",
      "Description: Homo sapiens phosphoglucomutase 1 (PGM1), transcript variant 1, mRNA\n",
      "Number of features: 44\n",
      "/molecule_type=mRNA\n",
      "/topology=linear\n",
      "/data_file_division=PRI\n",
      "/date=15-SEP-2018\n",
      "/accessions=['NM_002633']\n",
      "/sequence_version=2\n",
      "/keywords=['RefSeq']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='Asp263 missense variants perturb the active site of human phosphoglucomutase 1', ...), Reference(title='Induced Structural Disorder as a Molecular Mechanism for Enzyme Dysfunction in Phosphoglucomutase 1 Deficiency', ...), Reference(title='Phosphoglucomutase-1 deficiency: Intrafamilial clinical variability and common secondary adrenal insufficiency', ...), Reference(title='Compromised catalysis and potential folding defects in in vitro studies of missense mutants associated with hereditary phosphoglucomutase 1 deficiency', ...), Reference(title='Phosphoglucomutase1 is necessary for sustained cell growth under repetitive glucose depletion', ...), Reference(title='Congenital Disorders of N-Linked Glycosylation and Multiple Pathway Overview', ...), Reference(title='Treatment of Haemophilus aphrophilus endocarditis with ciprofloxacin', ...), Reference(title='Phosphoglucomutase 1: complete human and rabbit mRNA sequences and direct mapping of this highly polymorphic marker on human chromosome 1', ...), Reference(title='Antigenic analysis of the major human phosphoglucomutase isozymes: PGM1, PGM2, PGM3 and PGM4', ...), Reference(title='Gene localisation of the PGM1 enzyme system and the Duffy blood groups on chromosome No. 1 by means of a new fragile site at 1p31', ...)]\n",
      "/comment=REVIEWED REFSEQ: This record has been curated by NCBI staff. The\n",
      "reference sequence was derived from BC019920.1.\n",
      "This sequence is a reference standard in the RefSeqGene project.\n",
      "On Jun 10, 2002 this sequence version replaced NM_002633.1.\n",
      "Summary: The protein encoded by this gene is an isozyme of\n",
      "phosphoglucomutase (PGM) and belongs to the phosphohexose mutase\n",
      "family. There are several PGM isozymes, which are encoded by\n",
      "different genes and catalyze the transfer of phosphate between the\n",
      "1 and 6 positions of glucose. In most cell types, this PGM isozyme\n",
      "is predominant, representing about 90% of total PGM activity. In\n",
      "red cells, PGM2 is a major isozyme. This gene is highly\n",
      "polymorphic. Mutations in this gene cause glycogen storage disease\n",
      "type 14. Alternativley spliced transcript variants encoding\n",
      "different isoforms have been identified in this gene.[provided by\n",
      "RefSeq, Mar 2010].\n",
      "Transcript Variant: This variant (1) is dominant and encodes a\n",
      "protein of 562aa.\n",
      "Publication Note:  This RefSeq record includes a subset of the\n",
      "publications that are available for this gene. Please see the Gene\n",
      "record to access additional publications.\n",
      "                               [ECO:0000332]\n",
      "                               SAMEA1965299, SAMEA1966682\n",
      "                               [ECO:0000348]\n",
      "/structured_comment=OrderedDict([('Evidence-Data', OrderedDict([('Transcript exon combination', 'BC019920.1, SRR1803612.21972.1'), ('RNAseq introns', 'single sample supports all introns')]))])\n",
      "Seq('CCTTTCCCCTCCCGCCGGACCTGCCAGGAGGTGGGCTGGCGCGGAGGGAGGGCC...AAA', IUPACAmbiguousDNA())\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: NM_001172819.1\n",
      "Name: NM_001172819\n",
      "Description: Homo sapiens phosphoglucomutase 1 (PGM1), transcript variant 3, mRNA\n",
      "Number of features: 21\n",
      "/molecule_type=mRNA\n",
      "/topology=linear\n",
      "/data_file_division=PRI\n",
      "/date=24-JUN-2018\n",
      "/accessions=['NM_001172819']\n",
      "/sequence_version=1\n",
      "/keywords=['RefSeq']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='Asp263 missense variants perturb the active site of human phosphoglucomutase 1', ...), Reference(title='Induced Structural Disorder as a Molecular Mechanism for Enzyme Dysfunction in Phosphoglucomutase 1 Deficiency', ...), Reference(title='Phosphoglucomutase-1 deficiency: Intrafamilial clinical variability and common secondary adrenal insufficiency', ...), Reference(title='Compromised catalysis and potential folding defects in in vitro studies of missense mutants associated with hereditary phosphoglucomutase 1 deficiency', ...), Reference(title='Phosphoglucomutase1 is necessary for sustained cell growth under repetitive glucose depletion', ...), Reference(title='Congenital Disorders of N-Linked Glycosylation and Multiple Pathway Overview', ...), Reference(title='Treatment of Haemophilus aphrophilus endocarditis with ciprofloxacin', ...), Reference(title='Phosphoglucomutase 1: complete human and rabbit mRNA sequences and direct mapping of this highly polymorphic marker on human chromosome 1', ...), Reference(title='Antigenic analysis of the major human phosphoglucomutase isozymes: PGM1, PGM2, PGM3 and PGM4', ...), Reference(title='Gene localisation of the PGM1 enzyme system and the Duffy blood groups on chromosome No. 1 by means of a new fragile site at 1p31', ...)]\n",
      "/comment=REVIEWED REFSEQ: This record has been curated by NCBI staff. The\n",
      "reference sequence was derived from AK293298.1 and BC019920.1.\n",
      "Summary: The protein encoded by this gene is an isozyme of\n",
      "phosphoglucomutase (PGM) and belongs to the phosphohexose mutase\n",
      "family. There are several PGM isozymes, which are encoded by\n",
      "different genes and catalyze the transfer of phosphate between the\n",
      "1 and 6 positions of glucose. In most cell types, this PGM isozyme\n",
      "is predominant, representing about 90% of total PGM activity. In\n",
      "red cells, PGM2 is a major isozyme. This gene is highly\n",
      "polymorphic. Mutations in this gene cause glycogen storage disease\n",
      "type 14. Alternativley spliced transcript variants encoding\n",
      "different isoforms have been identified in this gene.[provided by\n",
      "RefSeq, Mar 2010].\n",
      "Transcript Variant: This variant (3) has an alternate 5' exon,\n",
      "which results in a downstream AUG start codon, as compared to\n",
      "variant 1. The resulting isoform (3) is shorter at the N-terminus,\n",
      "as compared to isoform 1.\n",
      "Publication Note:  This RefSeq record includes a subset of the\n",
      "publications that are available for this gene. Please see the Gene\n",
      "record to access additional publications.\n",
      "                               [ECO:0000332]\n",
      "                               SAMEA2145240, SAMEA2146236\n",
      "                               [ECO:0000348]\n",
      "/structured_comment=OrderedDict([('Evidence-Data', OrderedDict([('Transcript exon combination', 'AK293298.1, SRR1803615.2723.1'), ('RNAseq introns', 'single sample supports all introns')]))])\n",
      "Seq('GCTCGGGGCCGGCCGCTTCCGCGCGCTGCCGCCTCGGTTTCCACCTCCCGCTCC...AAA', IUPACAmbiguousDNA())\n",
      "ID: NG_016966.1\n",
      "Name: NG_016966\n",
      "Description: Homo sapiens phosphoglucomutase 1 (PGM1), RefSeqGene on chromosome 1\n",
      "Number of features: 24\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=PRI\n",
      "/date=29-MAR-2018\n",
      "/accessions=['NG_016966']\n",
      "/sequence_version=1\n",
      "/keywords=['RefSeq', 'RefSeqGene']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='Congenital Disorders of N-Linked Glycosylation and Multiple Pathway Overview', ...)]\n",
      "/comment=REVIEWED REFSEQ: This record has been curated by NCBI staff. The\n",
      "reference sequence was derived from AL109925.11.\n",
      "This sequence is a reference standard in the RefSeqGene project.\n",
      "Summary: The protein encoded by this gene is an isozyme of\n",
      "phosphoglucomutase (PGM) and belongs to the phosphohexose mutase\n",
      "family. There are several PGM isozymes, which are encoded by\n",
      "different genes and catalyze the transfer of phosphate between the\n",
      "1 and 6 positions of glucose. In most cell types, this PGM isozyme\n",
      "is predominant, representing about 90% of total PGM activity. In\n",
      "red cells, PGM2 is a major isozyme. This gene is highly\n",
      "polymorphic. Mutations in this gene cause glycogen storage disease\n",
      "type 14. Alternativley spliced transcript variants encoding\n",
      "different isoforms have been identified in this gene.[provided by\n",
      "RefSeq, Mar 2010].\n",
      "Seq('AAATACATTTCTTTAAATCATTGCAATTGGAATGCAAATGAAGGATAAATATGA...TGT', IUPACAmbiguousDNA())\n",
      "ID: DQ893786.2\n",
      "Name: DQ893786\n",
      "Description: Synthetic construct Homo sapiens clone IMAGE:100008246; FLH164382.01L; RZPDo839F09161D phosphoglucomutase 1 (PGM1) gene, encodes complete protein\n",
      "Number of features: 8\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=SYN\n",
      "/date=14-JUL-2016\n",
      "/accessions=['DQ893786']\n",
      "/sequence_version=2\n",
      "/keywords=['Human ORF Project', 'ORFeome collaboration', 'Gateway cloning system', 'full-length ORF without stop codon', 'FLEXGene']\n",
      "/source=synthetic construct\n",
      "/organism=synthetic construct\n",
      "/taxonomy=['other sequences', 'artificial sequences']\n",
      "/references=[Reference(title='Cloning of human full-length CDS FLEXGene in Gateway(TM) recombinational vector system', ...), Reference(title='Direct Submission', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=On Jan 22, 2007 this sequence version replaced DQ893786.1.\n",
      "This CDS clone is a part of a collection of human full-length ORF\n",
      "clones generated and verified by Harvard Institute of Proteomics\n",
      "(HIP) and Deutsches Ressourcenzentrum fuer Genomforschung GmbH\n",
      "(RZPD, Heubnerweg 6, D-14059 Berlin, Germany).  This ORF clone has\n",
      "been cloned without stop-codon (to allow fusion with C-terminal\n",
      "tag). AttB recombination sites have been added to either end of the\n",
      "ORF and directionally cloned using the Gateway cloning system into\n",
      "pDONR221. Additional sequences in the clone: 'ACC' before the 'ATG'\n",
      "(corresponding to Kozak consensus sequences).  Each clone is\n",
      "clonally isolated and full-length sequence-verified.  This clone is\n",
      "available for distribution at HIP (http://plasmid.hms.harvard.edu/)\n",
      "and RZPD (http://www.rzpd.de/products/orfclones/rzpdexpression/).\n",
      "This clone is part of the international ORFeome Collaboration\n",
      "(http://www.orfeomecollaboration.org/).\n",
      "Seq('GTACAAAAAAGCAGGCTCCACCATGGTGAAGATCGTGACAGTTAAGACCCAGGC...TAC', IUPACAmbiguousDNA())\n",
      "ID: DQ890618.2\n",
      "Name: DQ890618\n",
      "Description: Synthetic construct Homo sapiens clone IMAGE:100003248; FLH263531.01X; RZPDo839F10162D phosphoglucomutase 1 (PGM1) gene, encodes complete protein\n",
      "Number of features: 8\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=SYN\n",
      "/date=27-SEP-2007\n",
      "/accessions=['DQ890618']\n",
      "/sequence_version=2\n",
      "/keywords=['Human ORF Project', 'ORFeome collaboration', 'Gateway cloning system', 'full-length ORF with stop codon', 'FLEXGene']\n",
      "/source=synthetic construct\n",
      "/organism=synthetic construct\n",
      "/taxonomy=['other sequences', 'artificial sequences']\n",
      "/references=[Reference(title='Cloning of human full-length CDS FLEXGene in Gateway(TM) recombinational vector system', ...), Reference(title='Direct Submission', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=On Jan 25, 2007 this sequence version replaced DQ890618.1.\n",
      "This CDS clone is a part of a collection of human full-length ORF\n",
      "clones generated and verified by Harvard Institute of Proteomics\n",
      "(HIP) and Deutsches Ressourcenzentrum fuer Genomforschung GmbH\n",
      "(RZPD, Heubnerweg 6, D-14059 Berlin, Germany).  This ORF clone has\n",
      "been cloned with normalized stop-codon (tag). AttB recombination\n",
      "sites have been added to either end of the ORF and directionally\n",
      "cloned using the Gateway cloning system into pDONR221. Additional\n",
      "sequences in the clone: 'ACC' before the 'ATG' (corresponding to\n",
      "Kozak consensus sequences).  Each clone is clonally isolated and\n",
      "full-length sequence-verified.  This clone is available for\n",
      "distribution at HIP (http://plasmid.hms.harvard.edu/) and RZPD\n",
      "(http://www.rzpd.de/products/orfclones/rzpdexpression/).  This\n",
      "clone is part of the international ORFeome Collaboration\n",
      "(http://www.orfeomecollaboration.org/).\n",
      "Seq('GTACAAAAAAGCAGGCTCCACCATGGTGAAGATCGTGACAGTTAAGACCCAGGC...TAC', IUPACAmbiguousDNA())\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: AJ243265.1\n",
      "Name: AJ243265\n",
      "Description: Homo sapiens partial PGM1 gene for phosphoglucomutase 1, exons 5-7\n",
      "Number of features: 8\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=PRI\n",
      "/date=14-JUL-2016\n",
      "/accessions=['AJ243265']\n",
      "/sequence_version=1\n",
      "/keywords=['carbohydrate metabolism', 'PGM1 gene', 'phosphoglucomutase 1']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='', ...), Reference(title='Mapping recombination hotspots in human phosphoglucomutase (PGM1)', ...), Reference(title='Direct Submission', ...)]\n",
      "Seq('GGCTGGAGTGCAGTGGTGCAATCTCAGCTCACTGTAACCTCGCCTCCGAAGTTC...AGA', IUPACAmbiguousDNA())\n",
      "ID: CU676479.1\n",
      "Name: CU676479\n",
      "Description: Synthetic construct Homo sapiens gateway clone IMAGE:100019877 3' read PGM1 mRNA\n",
      "Number of features: 3\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=SYN\n",
      "/date=19-FEB-2008\n",
      "/accessions=['CU676479']\n",
      "/sequence_version=1\n",
      "/keywords=['Gateway cloning system']\n",
      "/source=synthetic construct\n",
      "/organism=synthetic construct\n",
      "/taxonomy=['other sequences', 'artificial sequences']\n",
      "/references=[Reference(title='Human ORFeome version 1.1: a platform for reverse proteomics', ...), Reference(title='hORFeome v3.1: a resource of human open reading frames representing over 10,000 human genes', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Clone structure of the ORF and flanking sequences is as follows:\n",
      "5'- g tac aaa aaa gtt gGC 5'ORF3' cca act ttc ttg tac - 3' Clone\n",
      "distribution: information can be found through the\n",
      "I.M.A.G.E.Consortium/LLNL at\n",
      "http://image.llnl.gov or Gene Services at\n",
      "http://www.geneservice.co.uk/home This clone is also a part of\n",
      "ORFeome collaboration clones\n",
      "(http://www.orfeomecollaboration.org/).\n",
      "Seq('GTACAAGAAAGTTGGGTAGGTGATGACAGTGGGTGCAGTGCGTCCCGTCCTCTC...CTT', IUPACAmbiguousDNA())\n",
      "ID: CU676478.1\n",
      "Name: CU676478\n",
      "Description: Synthetic construct Homo sapiens gateway clone IMAGE:100019877 5' read PGM1 mRNA\n",
      "Number of features: 3\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=SYN\n",
      "/date=19-FEB-2008\n",
      "/accessions=['CU676478']\n",
      "/sequence_version=1\n",
      "/keywords=['Gateway cloning system']\n",
      "/source=synthetic construct\n",
      "/organism=synthetic construct\n",
      "/taxonomy=['other sequences', 'artificial sequences']\n",
      "/references=[Reference(title='Human ORFeome version 1.1: a platform for reverse proteomics', ...), Reference(title='hORFeome v3.1: a resource of human open reading frames representing over 10,000 human genes', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Clone structure of the ORF and flanking sequences is as follows:\n",
      "5'- g tac aaa aaa gtt gGC 5'ORF3' cca act ttc ttg tac - 3' Clone\n",
      "distribution: information can be found through the\n",
      "I.M.A.G.E.Consortium/LLNL at\n",
      "http://image.llnl.gov or Gene Services at\n",
      "http://www.geneservice.co.uk/home This clone is also a part of\n",
      "ORFeome collaboration clones\n",
      "(http://www.orfeomecollaboration.org/).\n",
      "Seq('GTACAAAAAAGTTGGCATGGTGAAGATCGTGACAGTTAAGACCCAGGCGTACCA...ACA', IUPACAmbiguousDNA())\n",
      "ID: CH471059.2\n",
      "Name: CH471059\n",
      "Description: Homo sapiens 211000035844098 genomic scaffold, whole genome shotgun sequence\n",
      "Database cross-references: BioProject:PRJNA1431, BioSample:SAMN02981219\n",
      "Number of features: 1\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=CON\n",
      "/date=23-MAR-2015\n",
      "/accessions=['CH471059']\n",
      "/sequence_version=2\n",
      "/keywords=['WGS']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='The sequence of the human genome', ...), Reference(title='Direct Submission', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=On Sep 7, 2005 this sequence version replaced CH471059.1.\n",
      "This version of genes, transcripts, and proteins was created in\n",
      "April 2002 from the whole-genome mapping of transcript and protein\n",
      "sequences on the Celera human genome assembly\n",
      "(http://www.ncbi.nlm.nih.gov/genome/guide/human/release_notes.html)\n",
      "by Celera Chromosome Team, Content Systems and Informatics\n",
      "Research. The data sets used by this annotation process were\n",
      "collected in 2001 and include RefSeq (NM_) sequences, manually\n",
      "annotated transcripts from previous Celera assemblies, GenBank mRNA\n",
      "and dbEST sequences, Celera internal EST and full-insert sequences\n",
      "of cDNA clones (unpublished), mammalian SwissProt sequences, NRAA\n",
      "sequences, and International Protein Index (IPI) sequences (all\n",
      "human unless noted otherwise).  The CDS of each transcript was\n",
      "computationally defined by either the longest ATG-to-Stop or the\n",
      "longest open reading frame. All CDSs corresponding to the longest\n",
      "open reading frames with no starting ATG were flagged as partial.\n",
      "In addition, some of the genes were manually curated between 2002\n",
      "and 2005. Coverage analysis of splice junction donor/acceptor pairs\n",
      "and single-exon transcripts was done by Applied Biosystems in\n",
      "August 2006, based on Dec 2005 versions of human cDNA (RefSeq,\n",
      "Genbank mRNA and dbEST, Celera internal EST and full insert\n",
      "sequences) and human SwissProt sequences.\n",
      "This is the November 2001 combined whole genome shotgun assembly\n",
      "applied to the 27 million reads of Celera's whole genome shotgun\n",
      "data and 16 million  reads of shredded GenBank data from other\n",
      "human genome projects (Nature 2001. 409:860-921). It relied on\n",
      "Celera's paired reads and BAC end reads from TIGR for long range\n",
      "order and orientation. Its scaffolds were mapped to chromosomes\n",
      "using STS maps. For more detailed information about whole genome\n",
      "sequencing and Celera's assembly process, please refer to Venter,\n",
      "J.C. et al. Science 2001. 291:1304-1351.\n",
      "/contig=join(complement(AADB02000270.1:1..45802),gap(366),complement(AADB02000271.1:1..5785),gap(275),complement(AADB02000272.1:1..15280),gap(126),complement(AADB02000273.1:1..24961),gap(401),AADB02000274.1:1..1328,gap(1059),complement(AADB02000275.1:1..1072),gap(981),complement(AADB02000276.1:1..9162),gap(547),complement(AADB02000277.1:1..5792),gap(1443),AADB02000278.1:1..13071,gap(152),complement(AADB02000279.1:1..59478),gap(140),AADB02000280.1:1..3914,gap(197),complement(AADB02000281.1:1..396772),gap(20),complement(AADB02000282.1:1..329145),gap(25),complement(AADB02000283.1:1..7328),gap(209),complement(AADB02000284.1:1..59154),gap(20),complement(AADB02000285.1:1..247873),gap(241),AADB02000286.1:1..48277,gap(238),complement(AADB02000287.1:1..13874),gap(301),complement(AADB02000288.1:1..450362),gap(20),complement(AADB02000289.1:1..29490),gap(20),complement(AADB02000290.1:1..29580),gap(20),complement(AADB02000291.1:1..19580),gap(20),complement(AADB02000292.1:1..32655),gap(306),complement(AADB02000293.1:1..72291),gap(95),complement(AADB02000294.1:1..122818),gap(209),complement(AADB02000295.1:1..236),gap(1124),AADB02000296.1:1..9717,gap(20),complement(AADB02000297.1:1..7458),gap(87),complement(AADB02000298.1:1..7197),gap(20),complement(AADB02000299.1:1..1329),gap(158),complement(AADB02000300.1:1..2442),gap(20),AADB02000301.1:1..2532,gap(719),complement(AADB02000302.1:1..10841),gap(20),complement(AADB02000303.1:1..14747),gap(156),complement(AADB02000304.1:1..325094),gap(861),complement(AADB02000305.1:1..24359),gap(20),complement(AADB02000306.1:1..38162),gap(652),complement(AADB02000307.1:1..446200),gap(572),complement(AADB02000308.1:1..1406),gap(78),AADB02000309.1:1..7146,gap(191),AADB02000310.1:1..3261,gap(218),complement(AADB02000311.1:1..1015),gap(799),AADB02000312.1:1..3379,gap(88),AADB02000313.1:1..10169,gap(20),AADB02000314.1:1..4638,gap(199),complement(AADB02000315.1:1..762),gap(889),complement(AADB02000316.1:1..2484),gap(207),complement(AADB02000317.1:1..3713),gap(1095),AADB02000318.1:1..1451,gap(975),complement(AADB02000319.1:1..1389),gap(60),AADB02000320.1:1..4733,gap(244),complement(AADB02000321.1:1..7980),gap(1037),complement(AADB02000322.1:1..6754),gap(142),AADB02000323.1:1..9376,gap(193),complement(AADB02000324.1:1..6212),gap(91),complement(AADB02000325.1:1..7642),gap(186),complement(AADB02000326.1:1..17328),gap(20),AADB02000327.1:1..2741,gap(287),complement(AADB02000328.1:1..13215),gap(50),AADB02000329.1:1..11608,gap(20),complement(AADB02000330.1:1..109523),gap(20),AADB02000331.1:1..29178,gap(20),complement(AADB02000332.1:1..1223),gap(131),AADB02000333.1:1..6230,gap(20),AADB02000334.1:1..3926,gap(156),complement(AADB02000335.1:1..2262),gap(20),AADB02000336.1:1..11178,gap(2234),AADB02000337.1:1..1546,gap(39),AADB02000338.1:1..1177,gap(2426),complement(AADB02000339.1:1..51233),gap(51),AADB02000340.1:1..4380,gap(289),complement(AADB02000341.1:1..58408),gap(374),complement(AADB02000342.1:1..468713),gap(332),complement(AADB02000343.1:1..256240),gap(20),complement(AADB02000344.1:1..15874),gap(20),complement(AADB02000345.1:1..10739),gap(4119),complement(AADB02000346.1:1..9379),gap(42),complement(AADB02000347.1:1..2105),gap(20),AADB02000348.1:1..4318,gap(27),complement(AADB02000349.1:1..13403),gap(10365),complement(AADB02000350.1:1..123739),gap(3750),AADB02000351.1:1..12110,gap(20),complement(AADB02000352.1:1..3902),gap(732),complement(AADB02000353.1:1..43373),gap(537),complement(AADB02000354.1:1..1145321),gap(763),complement(AADB02000355.1:1..63399),gap(28),complement(AADB02000356.1:1..53385),gap(41),complement(AADB02000357.1:1..27682),gap(452),AADB02000358.1:1..5767,gap(20),complement(AADB02000359.1:1..17316),gap(1268),complement(AADB02000360.1:1..19142),gap(20),complement(AADB02000361.1:1..21305),gap(20),complement(AADB02000362.1:1..21526),gap(20),complement(AADB02000363.1:1..32343),gap(29),complement(AADB02000364.1:1..3700),gap(20),complement(AADB02000365.1:1..17542),gap(109),complement(AADB02000366.1:1..12399),gap(100),complement(AADB02000367.1:1..6378),gap(37),complement(AADB02000368.1:1..10828),gap(123),complement(AADB02000369.1:1..165842),gap(346),complement(AADB02000370.1:1..12757),gap(20),complement(AADB02000371.1:1..3171),gap(121),complement(AADB02000372.1:1..26295),gap(4439),complement(AADB02000373.1:1..52388),gap(45),AADB02000374.1:1..10205,gap(20),complement(AADB02000375.1:1..11428),gap(20),complement(AADB02000376.1:1..34462),gap(495),complement(AADB02000377.1:1..21436),gap(20),complement(AADB02000378.1:1..246193),gap(20),complement(AADB02000379.1:1..14666),gap(249),complement(AADB02000380.1:1..116313),gap(131),complement(AADB02000381.1:1..292273),gap(20),complement(AADB02000382.1:1..34772),gap(20),AADB02000383.1:1..1788,gap(349),complement(AADB02000384.1:1..103580),gap(39),AADB02000385.1:1..63430,gap(197),complement(AADB02000386.1:1..4192),gap(98),complement(AADB02000387.1:1..15376),gap(68),complement(AADB02000388.1:1..125291),gap(572),complement(AADB02000389.1:1..59606),gap(1319),complement(AADB02000390.1:1..4235),gap(20),complement(AADB02000391.1:1..15967),gap(20),complement(AADB02000392.1:1..90334),gap(281),AADB02000393.1:1..24180,gap(20),complement(AADB02000394.1:1..12550),gap(188),complement(AADB02000395.1:1..197051),gap(49),complement(AADB02000396.1:1..169154),gap(204),AADB02000397.1:1..102633,gap(421),AADB02000398.1:1..50748,gap(514),AADB02000399.1:1..41806,gap(141),AADB02000400.1:1..29483,gap(102),AADB02000401.1:1..136862,gap(20),complement(AADB02000402.1:1..476065),gap(20),complement(AADB02000403.1:1..574975),gap(5272),complement(AADB02000404.1:1..396737),gap(350),complement(AADB02000405.1:1..54777),gap(20),complement(AADB02000406.1:1..38888),gap(328),complement(AADB02000407.1:1..25999),gap(20),AADB02000408.1:1..270,gap(20),complement(AADB02000409.1:1..67988),gap(2231),complement(AADB02000410.1:1..242015),gap(20),AADB02000411.1:1..24003,gap(1407),complement(AADB02000412.1:1..153746),gap(423),complement(AADB02000413.1:1..32330),gap(20),complement(AADB02000414.1:1..515334),gap(389),complement(AADB02000415.1:1..67553),gap(20),complement(AADB02000416.1:1..38426),gap(20),complement(AADB02000417.1:1..10300),gap(1122),complement(AADB02000418.1:1..3193),gap(20),complement(AADB02000419.1:1..153026),gap(1539),complement(AADB02000420.1:1..103779),gap(610),complement(AADB02000421.1:1..2697),gap(124),complement(AADB02000422.1:1..625053),gap(20),complement(AADB02000423.1:1..479102),gap(274),complement(AADB02000424.1:1..42305),gap(20),AADB02000425.1:1..33970,gap(436),complement(AADB02000426.1:1..229366),gap(20),AADB02000427.1:1..18335,gap(170),complement(AADB02000428.1:1..15130),gap(98),complement(AADB02000429.1:1..3003),gap(41),complement(AADB02000430.1:1..8993),gap(1541),complement(AADB02000431.1:1..29932),gap(20),complement(AADB02000432.1:1..9548),gap(199),AADB02000433.1:1..2499,gap(70),AADB02000434.1:1..2328,gap(587),complement(AADB02000435.1:1..939491),gap(453),AADB02000436.1:1..1595,gap(53),AADB02000437.1:1..3525,gap(20),AADB02000438.1:1..620,gap(161),complement(AADB02000439.1:1..2245),gap(241),complement(AADB02000440.1:1..10725),gap(363),AADB02000441.1:1..4305,gap(185),complement(AADB02000442.1:1..16086),gap(20),complement(AADB02000443.1:1..3923),gap(803),complement(AADB02000444.1:1..29674),gap(20),AADB02000445.1:1..6269,gap(272),complement(AADB02000446.1:1..5550),gap(20),complement(AADB02000447.1:1..11385),gap(504),complement(AADB02000448.1:1..296983),gap(20),complement(AADB02000449.1:1..216467),gap(311),AADB02000450.1:1..21635,gap(20),complement(AADB02000451.1:1..191127),gap(20),complement(AADB02000452.1:1..14043),gap(3083),complement(AADB02000453.1:1..151425),gap(20),complement(AADB02000454.1:1..43294),gap(20),AADB02000455.1:1..1470,gap(253),AADB02000456.1:1..10482,gap(20),complement(AADB02000457.1:1..17977),gap(370),complement(AADB02000458.1:1..6405),gap(84),complement(AADB02000459.1:1..3428),gap(242),AADB02000460.1:1..4828,gap(71),complement(AADB02000461.1:1..132207),gap(119),complement(AADB02000462.1:1..8863),gap(20),AADB02000463.1:1..1238,gap(20),complement(AADB02000464.1:1..25422),gap(311),AADB02000465.1:1..532738,gap(20),complement(AADB02000466.1:1..5216),gap(20),complement(AADB02000467.1:1..3611),gap(673),complement(AADB02000468.1:1..46016),gap(411),AADB02000469.1:1..90168,gap(220),complement(AADB02000470.1:1..38369),gap(716),AADB02000471.1:1..28770,gap(235),complement(AADB02000472.1:1..21527),gap(222),complement(AADB02000473.1:1..5999),gap(20),complement(AADB02000474.1:1..318807),gap(317),complement(AADB02000475.1:1..71945),gap(604),complement(AADB02000476.1:1..7013),gap(1338),complement(AADB02000477.1:1..11522),gap(1335),complement(AADB02000478.1:1..42702),gap(732),complement(AADB02000479.1:1..756078),gap(67),complement(AADB02000480.1:1..430933),gap(75),complement(AADB02000481.1:1..68374),gap(20),complement(AADB02000482.1:1..134159),gap(74),AADB02000483.1:1..87824,gap(20),AADB02000484.1:1..82566,gap(20),complement(AADB02000485.1:1..45751),gap(20),AADB02000486.1:1..74367,gap(167),complement(AADB02000487.1:1..346284),gap(168),complement(AADB02000488.1:1..46850),gap(661),AADB02000489.1:1..758,gap(135),complement(AADB02000490.1:1..28595),gap(249),AADB02000491.1:1..115703,gap(58),complement(AADB02000492.1:1..257287),gap(20),AADB02000493.1:1..3037,gap(20),AADB02000494.1:1..3777,gap(1829),AADB02000495.1:1..927993,gap(24),AADB02000496.1:1..181696,gap(152),complement(AADB02000497.1:1..251290),gap(20),complement(AADB02000498.1:1..104752),gap(208),AADB02000499.1:1..327553,gap(20),complement(AADB02000500.1:1..6074),gap(311),complement(AADB02000501.1:1..1678),gap(306),AADB02000502.1:1..3622,gap(20),complement(AADB02000503.1:1..10876),gap(1677),complement(AADB02000504.1:1..1456),gap(20),complement(AADB02000505.1:1..1758),gap(145),complement(AADB02000506.1:1..2976),gap(1102),complement(AADB02000507.1:1..2094),gap(609),complement(AADB02000508.1:1..413066),gap(20),complement(AADB02000509.1:1..49048),gap(20),AADB02000510.1:1..4000,gap(88),complement(AADB02000511.1:1..40008),gap(20),complement(AADB02000512.1:1..639065),gap(7671),complement(AADB02000513.1:1..23236),gap(20),AADB02000514.1:1..9103,gap(320),complement(AADB02000515.1:1..108444),gap(145),AADB02000516.1:1..4357,gap(20),complement(AADB02000517.1:1..44219),gap(4853),complement(AADB02000518.1:1..56682),gap(6821),complement(AADB02000519.1:1..30286),gap(20),AADB02000520.1:1..237519,gap(154),complement(AADB02000521.1:1..864480),gap(10403),complement(AADB02000522.1:1..1763751),gap(20),complement(AADB02000523.1:1..608666),gap(229),complement(AADB02000524.1:1..21853),gap(347),complement(AADB02000525.1:1..43038),gap(20),AADB02000526.1:1..181968,gap(20),AADB02000527.1:1..1073823,gap(38),complement(AADB02000528.1:1..246597),gap(297),complement(AADB02000529.1:1..100007),gap(20),complement(AADB02000530.1:1..285101),gap(195),complement(AADB02000531.1:1..8918),gap(293),AADB02000532.1:1..7001,gap(53),AADB02000533.1:1..6384,gap(262),AADB02000534.1:1..133822,gap(409),complement(AADB02000535.1:1..10283),gap(20),complement(AADB02000536.1:1..74648),gap(203),complement(AADB02000537.1:1..33824),gap(1908),AADB02000538.1:1..40981,gap(314),AADB02000539.1:1..392335,gap(20),complement(AADB02000540.1:1..174547),gap(462),complement(AADB02000541.1:1..22459),gap(54),AADB02000542.1:1..320760,gap(20),complement(AADB02000543.1:1..473375),gap(20),complement(AADB02000544.1:1..148243),gap(365),complement(AADB02000545.1:1..65676),gap(177),complement(AADB02000546.1:1..469009),gap(20),complement(AADB02000547.1:1..1371810),gap(114),AADB02000548.1:1..4193,gap(151),AADB02000549.1:1..29564,gap(552),AADB02000550.1:1..2448,gap(78),AADB02000551.1:1..16384,gap(168),AADB02000552.1:1..34432,gap(398),complement(AADB02000553.1:1..652917),gap(20),complement(AADB02000554.1:1..534056),gap(20),AADB02000555.1:1..6223,gap(284),complement(AADB02000556.1:1..7865),gap(67),complement(AADB02000557.1:1..1227672),gap(20),complement(AADB02000558.1:1..8545),gap(20),complement(AADB02000559.1:1..59841),gap(613),AADB02000560.1:1..36093,gap(20),complement(AADB02000561.1:1..645612),gap(20),AADB02000562.1:1..5095,gap(302),complement(AADB02000563.1:1..1247),gap(20),complement(AADB02000564.1:1..1398),gap(419),complement(AADB02000565.1:1..10695),gap(577),AADB02000566.1:1..35990,gap(134),complement(AADB02000567.1:1..23717),gap(20),complement(AADB02000568.1:1..200039),gap(20),complement(AADB02000569.1:1..1205032),gap(145),complement(AADB02000570.1:1..578703),gap(937),AADB02000571.1:1..164993,gap(20),complement(AADB02000572.1:1..545682),gap(20),complement(AADB02000573.1:1..117299),gap(208),complement(AADB02000574.1:1..316281),gap(20),complement(AADB02000575.1:1..294508),gap(20),complement(AADB02000576.1:1..94570),gap(20),complement(AADB02000577.1:1..229659),gap(237),complement(AADB02000578.1:1..3353),gap(2050),complement(AADB02000579.1:1..910736),gap(715),complement(AADB02000580.1:1..31153),gap(377),complement(AADB02000581.1:1..323006),gap(302),complement(AADB02000582.1:1..327458),gap(270),AADB02000583.1:1..4065,gap(41),complement(AADB02000584.1:1..297047),gap(98),AADB02000585.1:1..11956,gap(484),complement(AADB02000586.1:1..40390),gap(57),AADB02000587.1:1..16091,gap(20),complement(AADB02000588.1:1..1292137),gap(149),complement(AADB02000589.1:1..781582),gap(20),complement(AADB02000590.1:1..776460),gap(512),complement(AADB02000591.1:1..3845239),gap(20),complement(AADB02000592.1:1..165346),gap(2879),complement(AADB02000593.1:1..597008),gap(774),complement(AADB02000594.1:1..15020),gap(233),complement(AADB02000595.1:1..16650),gap(20),complement(AADB02000596.1:1..21962),gap(20),complement(AADB02000597.1:1..127910),gap(1901),complement(AADB02000598.1:1..174237),gap(20),complement(AADB02000599.1:1..122129),gap(344),complement(AADB02000600.1:1..1020950),gap(20),complement(AADB02000601.1:1..499527),gap(288),complement(AADB02000602.1:1..147959),gap(20),complement(AADB02000603.1:1..1210473),gap(20),complement(AADB02000604.1:1..186174),gap(213),complement(AADB02000605.1:1..792893),gap(20),complement(AADB02000606.1:1..427133),gap(20),AADB02000607.1:1..15585,gap(20),AADB02000608.1:1..36598,gap(1087),AADB02000609.1:1..487,gap(111),AADB02000610.1:1..4169,gap(237),AADB02000611.1:1..71109,gap(20),complement(AADB02000612.1:1..9948),gap(20),complement(AADB02000613.1:1..125355),gap(20),AADB02000614.1:1..1468,gap(20),complement(AADB02000615.1:1..492607),gap(20),complement(AADB02000616.1:1..210961),gap(177),complement(AADB02000617.1:1..158292),gap(5311),complement(AADB02000618.1:1..981531),gap(20),complement(AADB02000619.1:1..1105931),gap(2420),complement(AADB02000620.1:1..557263),gap(524),complement(AADB02000621.1:1..526061),gap(67),complement(AADB02000622.1:1..1200),gap(20),AADB02000623.1:1..5894,gap(268),AADB02000624.1:1..22933,gap(67),AADB02000625.1:1..5054,gap(20),AADB02000626.1:1..14482,gap(752),AADB02000627.1:1..3893,gap(140),complement(AADB02000628.1:1..1100350))\n",
      "UnknownSeq(57745789, alphabet=IUPACAmbiguousDNA(), character='N')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: BC067763.1\n",
      "Name: BC067763\n",
      "Description: Homo sapiens phosphoglucomutase 1, mRNA (cDNA clone MGC:87068 IMAGE:5286530), complete cds\n",
      "Number of features: 8\n",
      "/molecule_type=mRNA\n",
      "/topology=linear\n",
      "/data_file_division=PRI\n",
      "/date=19-MAR-2007\n",
      "/accessions=['BC067763']\n",
      "/sequence_version=1\n",
      "/keywords=['MGC']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='Generation and initial analysis of more than 15,000 full-length human and mouse cDNA sequences', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Contact: MGC help desk\n",
      "Email: cgapbs-r@mail.nih.gov\n",
      "Tissue Procurement: Miklos Palkovits, M.D., Ph.D.\n",
      "cDNA Library Preparation: Michael J. Brownstein (NHGRI) &  Shiraki\n",
      "Toshiyuki and Piero Carninci (RIKEN)\n",
      "cDNA Library Arrayed by: The I.M.A.G.E. Consortium (LLNL)\n",
      "DNA Sequencing by: Sequencing Group at the Stanford Human Genome\n",
      "Center, Stanford University School of Medicine, Stanford, CA  94305\n",
      "Web site:       http://www-shgc.stanford.edu\n",
      "Contact:  (Dickson, Mark) mcd@paxil.stanford.edu\n",
      "Dickson, M., Schmutz, J., Grimwood, J., Rodriquez, A., and Myers,\n",
      "R. M.\n",
      "Clone distribution: MGC clone distribution information can be found\n",
      "through the I.M.A.G.E. Consortium/LLNL at: http://image.llnl.gov\n",
      "Series: IRAK Plate: 168 Row: b Column: 18\n",
      "This clone was selected for full length sequencing because it\n",
      "passed the following selection criteria: matched mRNA gi: 21361620.\n",
      "Differences found between this sequence and the human reference\n",
      "genome (build 36) are described in misc_difference features below\n",
      "and these differences were also compared to chimpanzee genome\n",
      "(build 2).\n",
      "Seq('GTGCCCTCACCCCAGAGCAGCTGCAGCCTCAGCCGGCCGCCCCTCCGCCAGCCA...AAA', IUPACAmbiguousDNA())\n",
      "ID: BC001756.2\n",
      "Name: BC001756\n",
      "Description: Homo sapiens phosphoglucomutase 1, mRNA (cDNA clone MGC:1682 IMAGE:3505510), complete cds\n",
      "Number of features: 9\n",
      "/molecule_type=mRNA\n",
      "/topology=linear\n",
      "/data_file_division=PRI\n",
      "/date=19-MAR-2007\n",
      "/accessions=['BC001756']\n",
      "/sequence_version=2\n",
      "/keywords=['MGC']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='Generation and initial analysis of more than 15,000 full-length human and mouse cDNA sequences', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=On Aug 19, 2003 this sequence version replaced BC001756.1.\n",
      "Contact: MGC help desk\n",
      "Email: cgapbs-r@mail.nih.gov\n",
      "Tissue Procurement: ATCC\n",
      "cDNA Library Preparation: Rubin Laboratory\n",
      "cDNA Library Arrayed by: The I.M.A.G.E. Consortium (LLNL)\n",
      "DNA Sequencing by: Institute for Systems Biology\n",
      "http://www.systemsbiology.org\n",
      "contact: amadan@systemsbiology.org\n",
      "Anup Madan, Jessica Fahey, Erin Helton, Mark Ketteman, Anuradha\n",
      "Madan, Stephanie Rodrigues, Amy Sanchez and Michelle Whiting\n",
      "Clone distribution: MGC clone distribution information can be found\n",
      "through the I.M.A.G.E. Consortium/LLNL at: http://image.llnl.gov\n",
      "Series: IRAL Plate: 7 Row: b Column: 11\n",
      "This clone was selected for full length sequencing because it\n",
      "passed the following selection criteria: matched mRNA gi: 21361620.\n",
      "Differences found between this sequence and the human reference\n",
      "genome (build 36) are described in misc_difference features below\n",
      "and these differences were also compared to chimpanzee genome\n",
      "(build 2).\n",
      "Seq('CGCCGGCCTAGAGTGCGGCGTAGCCCCCACCCGCCGTGCCCTCACCCCAGAGCA...AAA', IUPACAmbiguousDNA())\n",
      "ID: BC019920.1\n",
      "Name: BC019920\n",
      "Description: Homo sapiens phosphoglucomutase 1, mRNA (cDNA clone MGC:29909 IMAGE:5122261), complete cds\n",
      "Number of features: 4\n",
      "/molecule_type=mRNA\n",
      "/topology=linear\n",
      "/data_file_division=PRI\n",
      "/date=15-JUL-2006\n",
      "/accessions=['BC019920']\n",
      "/sequence_version=1\n",
      "/keywords=['MGC']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='Generation and initial analysis of more than 15,000 full-length human and mouse cDNA sequences', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Contact: MGC help desk\n",
      "Email: cgapbs-r@mail.nih.gov\n",
      "Tissue Procurement: ATCC\n",
      "cDNA Library Preparation: Life Technologies, Inc.\n",
      "cDNA Library Arrayed by: The I.M.A.G.E. Consortium (LLNL)\n",
      "DNA Sequencing by: Baylor College of Medicine Human Genome\n",
      "Sequencing Center\n",
      "Center code: BCM-HGSC\n",
      "Web site: http://www.hgsc.bcm.tmc.edu/cdna/\n",
      "Contact: amg@bcm.tmc.edu\n",
      "Gunaratne, P.H., Garcia, A.M., Lu, X., Hulyk, S.W., Loulseged, H.,\n",
      "Kowis, C.R., Sneed, A.J., Martin, R.G., Muzny, D.M., Nanavati,\n",
      "A.N., Gibbs, R.A.\n",
      "Clone distribution: MGC clone distribution information can be found\n",
      "through the I.M.A.G.E. Consortium/LLNL at: http://image.llnl.gov\n",
      "Series: IRAK Plate: 42 Row: n Column: 18\n",
      "This clone was selected for full length sequencing because it\n",
      "passed the following selection criteria: matched mRNA gi: 21361620.\n",
      "Differences found between this sequence and the human reference\n",
      "genome (build 36) are described in misc_difference features below.\n",
      "Seq('CCTTTCCCCTCCCGCCGGACCTGCCAGGAGGTGGGCTGGCGCGGAGGGAGGGCC...AAA', IUPACAmbiguousDNA())\n",
      "ID: NC_000001.11\n",
      "Name: NC_000001\n",
      "Description: Homo sapiens chromosome 1, GRCh38.p12 Primary Assembly\n",
      "Database cross-references: BioProject:PRJNA168, Assembly:GCF_000001405.38\n",
      "Number of features: 1\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=CON\n",
      "/date=26-MAR-2018\n",
      "/accessions=['NC_000001']\n",
      "/sequence_version=11\n",
      "/keywords=['RefSeq']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='The DNA sequence and biological annotation of human chromosome 1', ...), Reference(title='Finishing the euchromatic sequence of the human genome', ...), Reference(title='Initial sequencing and analysis of the human genome', ...)]\n",
      "/comment=REFSEQ INFORMATION: The reference sequence is identical to\n",
      "CM000663.2.\n",
      "On Feb 3, 2014 this sequence version replaced NC_000001.10.\n",
      "Assembly Name: GRCh38.p12 Primary Assembly\n",
      "The DNA sequence is composed of genomic sequence, primarily\n",
      "finished clones that were sequenced as part of the Human Genome\n",
      "Project. PCR products and WGS shotgun sequence have been added\n",
      "where necessary to fill gaps or correct errors. All such additions\n",
      "are manually curated by GRC staff. For more information see:\n",
      "https://genomereference.org.\n",
      "                               pipeline\n",
      "/structured_comment=OrderedDict([('Genome-Annotation-Data', OrderedDict([('Annotation Provider', 'NCBI'), ('Annotation Status', 'Full annotation'), ('Annotation Name', 'Homo sapiens Annotation Release 109'), ('Annotation Version', '109'), ('Annotation Pipeline', 'NCBI eukaryotic genome annotation'), ('Annotation Software Version', '8.0'), ('Annotation Method', 'Best-placed RefSeq; Gnomon'), ('Features Annotated', 'Gene; mRNA; CDS; ncRNA')]))])\n",
      "/contig=join(gap(10000),NT_077402.3:1..197666,gap(50000),NT_187170.1:1..40302,gap(50000),NT_077912.2:1..188020,gap(50000),NT_032977.10:1..121390471,gap(50000),NT_187171.1:1..198076,gap(100),NT_187172.1:1..278512,gap(100),NT_187173.1:1..2282185,gap(100),NT_187174.1:1..63597,gap(100),NT_187175.1:1..83495,gap(100),NT_187176.1:1..251763,gap(18000000),NT_004487.20:1..80374348,gap(50000),NT_167186.2:1..25337487,gap(10000))\n",
      "UnknownSeq(248956422, alphabet=IUPACAmbiguousDNA(), character='N')\n",
      "ID: CM000252.1\n",
      "Name: CM000252\n",
      "Description: Homo sapiens chromosome 1, whole genome shotgun sequence\n",
      "Database cross-references: BioProject:PRJNA1431, BioSample:SAMN02981219\n",
      "Number of features: 1\n",
      "/molecule_type=DNA\n",
      "/topology=linear\n",
      "/data_file_division=CON\n",
      "/date=23-MAR-2015\n",
      "/accessions=['CM000252', 'AADB02000000', 'CH003472']\n",
      "/sequence_version=1\n",
      "/keywords=['WGS']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='The sequence of the human genome', ...), Reference(title='Direct Submission', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=On Sep 8, 2005 this sequence version replaced CH003472.1.\n",
      "This is the November 2001 combined whole genome shotgun assembly\n",
      "applied to the 27 million reads of Celera's whole genome shotgun\n",
      "data and 16 million  reads of shredded GenBank data from other\n",
      "human genome projects (Nature 2001. 409:860-921). It relied on\n",
      "Celera's paired reads and BAC end reads from TIGR for long range\n",
      "order and orientation. Its scaffolds were mapped to chromosomes\n",
      "using STS maps. For more detailed information about whole genome\n",
      "sequencing and Celera's assembly process, please refer to Venter,\n",
      "J.C. et al. Science 2001. 291:1304-1351.\n",
      "/contig=join(CH471183.1:1..1820082,gap(55678),CH471248.1:1..499236,gap(55678),CH471130.1:1..9603050,gap(55678),CH878695.1:1..29324,gap(55678),CH471167.1:1..3161209,gap(55678),CH471328.1:1..18102,gap(55678),CH878713.1:1..2808,gap(55678),CH471134.1:1..8495160,gap(55678),CH878410.1:1..9069,gap(55678),CH471059.2:1..57745789,gap(55678),CH471303.1:1..56688,gap(55678),CH878323.1:1..16553,gap(55678),CH878257.1:1..12222,gap(55678),CH471097.2:1..20146335,gap(55678),CH878622.1:1..18875,gap(55678),CH471156.1:1..4472039,gap(55678),CH878567.1:1..1100,gap(55678),CH471296.1:1..79913,gap(55678),CH878664.1:1..36353,gap(55678),CH471122.1:1..11481316,gap(55678),CH471286.1:1..132653,gap(55678),CH878248.1:1..1315,gap(55678),CH878369.1:1..2455,gap(55678),CH878566.1:1..1100,gap(55678),CH878319.1:1..1378,gap(55678),CH878450.1:1..1100,gap(55678),CH878386.1:1..3384,gap(55678),CH878594.1:1..1373,gap(55678),CH878349.1:1..3394,gap(55678),CH878325.1:1..7015,gap(55678),CH878559.1:1..1100,gap(55678),CH471244.1:1..523665,gap(55678),CH471254.1:1..444108,gap(55678),CH878643.1:1..11038,gap(55678),CH878669.1:1..6138,gap(55678),CH878711.1:1..22468,gap(55678),CH878263.1:1..6924,gap(55678),CH878651.1:1..4636,gap(55678),CH471275.1:1..175790,gap(55678),CH471269.1:1..222081,gap(55678),CH878646.1:1..22471,gap(55678),CH471223.1:1..908887,gap(55678),CH878687.1:1..8706,gap(55678),CH878475.1:1..4439,gap(55678),CH878249.1:1..3102,gap(55678),CH878725.1:1..23443,gap(55678),CH878271.1:1..6556,gap(55678),CH878618.1:1..34688,gap(55678),CH471121.2:1..11514521,gap(55678),CH878584.1:1..1100,gap(55678),CH878401.1:1..2253,gap(55678),CH878684.1:1..41481,gap(55678),CH471067.1:1..44521323,gap(55678),CH878423.1:1..1375,gap(55678),CH878443.1:1..2362,gap(55678),CH878552.1:1..1100,gap(55678),CH878381.1:1..5862,gap(55678),CH471268.1:1..222909,gap(55678),CH471100.2:1..17411188,gap(55678),complement(CH878588.1:1..1101),gap(55678),CH878506.1:1..5267,gap(55678),CH471098.1:1..18989345,gap(55678),CH878612.1:1..4121,gap(55678),CH878377.1:1..3579,gap(55678),CH878265.1:1..4299,gap(55678),CH471148.1:1..5334993,gap(55678),CH471257.1:1..355430)\n",
      "UnknownSeq(222388987, alphabet=IUPACAmbiguousDNA(), character='N')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: AK312254.1\n",
      "Name: AK312254\n",
      "Description: Homo sapiens cDNA, FLJ92549, highly similar to Homo sapiens phosphoglucomutase 1 (PGM1), mRNA\n",
      "Number of features: 2\n",
      "/molecule_type=mRNA\n",
      "/topology=linear\n",
      "/data_file_division=HTC\n",
      "/date=24-MAY-2008\n",
      "/accessions=['AK312254']\n",
      "/sequence_version=1\n",
      "/keywords=['HTC', 'HTC_FLI', 'oligo capping']\n",
      "/source=Homo sapiens (human)\n",
      "/organism=Homo sapiens\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      "/references=[Reference(title='NEDO functional analysis of protein and research application project', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Human cDNA sequencing project focused on splicing variants of mRNA\n",
      "in NEDO functional analysis of protein and research application\n",
      "project supported by Ministry of Economy, Trade and Industry,\n",
      "Japan; cDNA selection for complete cds sequencing: Reverse\n",
      "Proteomics Research Institute (REPRORI), Hitachi, Ltd., Japan\n",
      "(Hitachi) and Japan Biological Informatics Consortium, Japan\n",
      "(JBIC); cDNA complete cds sequencing: JBIC and Biological\n",
      "Information Research Center (BIRC), AIST; cDNA library\n",
      "construction: Helix Research Institute supported by Japan Key\n",
      "Technology Center, Japan (HRI); cDNA 5'- & 3'-end sequencing:\n",
      "Research Association for Biotechnology, Japan, Biotechnology\n",
      "Center, National Institute of Technology and Evaluation, Japan and\n",
      "HRI; cDNA mapping to human genome: Central Research Laboratory,\n",
      "Hitachi; evaluation and annotation: REPRORI.\n",
      "Seq('AGAGCAGCTGCAGCCTCAGCCGGCCGCCCCTCCGCCAGCCAAGTCCGCCGCTCT...TAA', IUPACAmbiguousDNA())\n"
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "from Bio import Entrez\n",
    "Entrez.email = 'joanne91218@berkeley.edu'\n",
    "\n",
    "\n",
    "gene=[]\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'homo sapiens[ORGN] PGM1',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp= SeqIO.read(handle,'gb')\n",
    "    gene_list= temp.annotation['db_source']\n",
    "    accession= gene_list.split()\n",
    "    gene.append(accession[len(accession)-1])\n",
    "print(gene)\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Creating tables and inserting data for genes, enzymes and pathways\n",
    "\n",
    "Gene Table:\tStart with name, description, organism, and nucleotide sequence. Additional fields might include chromosome, start and end position, strand, and translated sequence.\n",
    "\n",
    "Enzyme\tTable:\tIncludes name, function, and enzyme\tcommission (EC)\tnumber.\t\n",
    "\n",
    "Pathway\tTable: What\tfields belong in this table? Perhaps just name and description.\t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Making GENES table\n",
    "c.execute(\"\"\"DROP TABLE GENES\"\"\") # to prevent the table repeatedly create or insert data after running execution each time, a drop command is used\n",
    "c.execute(\"\"\"CREATE TABLE GENES (name TEXT,\n",
    "                                description TEXT,\n",
    "                                organism TEXT,\n",
    "                                nucleotide sequences TEXT,\n",
    "                                chromosome position TEXT,\n",
    "                                enzymes encoded TEXT);\"\"\")      # creating a table named GENES\n",
    "conn.commit()   #commit changes to the database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Inserting data into GENES table\n",
    "c.execute(\"\"\"INSERT INTO GENES    VALUES('')\n",
    "                                  \"\"\")\n",
    "\n",
    "conn.commit()\n",
    "                     "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Making ENZYMES table\n",
    "c.execute(\"\"\"DROP TABLE ENZYMES\"\"\")\n",
    "c.execute(\"\"\"CREATE TABLE ENZYMES (name TEXT,\n",
    "                                 function TEXT,\n",
    "                                 Enzyme commission number TEXT,\n",
    "                                 Pathway TEXT);\"\"\")\n",
    "conn.commit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Inserting data into ENZYMES table\n",
    "c.execute(\"\"\"INSERT INTO ENZYMES  VALUES('phosphoglucomutase',\n",
    "                                  'transfers a phosphate group on an -D-glucose monomer from the 1 to the 6 position',\n",
    "                                  'EC:5.4.2.2',\n",
    "                                  'glycolysis'),\n",
    "                                  \n",
    "                                  ('Aldose 1-epimerase',\n",
    "                                  'catalyzes the chemical reaction between alpha-D-glucose and beta-D-glucose',\n",
    "                                  'EC:5.1.3.3',\n",
    "                                  'glycolysis'),\n",
    "                                  \n",
    "                                  ('Enolase',\n",
    "                                  'catalyze the conversion of 2-phosphoglycerate(2-PG) to phosphoenolpyruvate(PEP)',\n",
    "                                  'EC:4.2.1.11',\n",
    "                                  'glycolysis'),\n",
    "                                  \n",
    "                                  ('Glucose-6-phosphate isomerase',\n",
    "                                  'interconvert glucose 6-phosphate and fructose 6-phosphate',\n",
    "                                  'EC:5.3.1.9',\n",
    "                                  'glycolysis'),\n",
    "                                  \n",
    "                                  ('Glucose-6-phosphate isomerase',\n",
    "                                  'interconvert glucose 6-phosphate and fructose 6-phosphate',\n",
    "                                  'EC:5.3.1.9',\n",
    "                                  'pentose phosphate pathway'),\n",
    "                                  \n",
    "                                  ('6-phosphofructokinase',\n",
    "                                  'convert fructose-6-phosphate into fructose 1,6-bisphosphate',\n",
    "                                  'EC:2.7.1.11',\n",
    "                                  'pentose phosphate pathway'),\n",
    "                                  \n",
    "                                  ('Fructose-bisphosphate aldolase',\n",
    "                                  'splits the aldol and fructose 1,6-bisphosphate into the triose phosphates dihydroxyacetone phosphate(DHAP) and glyceraldehyde 3-phosphate(G3P)',\n",
    "                                  'EC:4.1.2.13',\n",
    "                                  'pentose phosphate pathway'),\n",
    "                                  \n",
    "                                  ('Fructose-1,6-bisphosphatase',\n",
    "                                  'converts fructose-1,6-bisphosphate to fructose 6-phosphate',\n",
    "                                  'EC:3.1.3.11',\n",
    "                                  'pentose phosphate pathway'),\n",
    "                                  \n",
    "                                  ('pyruvate dehydrogenase E1 component',\n",
    "                                  'pyruvate decarboxylation and reductive acetylation of lipoic acid',\n",
    "                                  'EC:1.2.4.1',\n",
    "                                  'citric acid cycle'),\n",
    "                                  \n",
    "                                  ('pyruvate dehydrogenase E2 component',\n",
    "                                  'bind on lipoate-thioester and by transacylation produce acetyl-CoA',\n",
    "                                  'EC:2.3.1.12',\n",
    "                                  'citric acid cycle'),\n",
    "                                  \n",
    "                                  ('dihydrolipoamide dehydrogenase',\n",
    "                                  'oxidize dihydrolipoamide to lipoamide',\n",
    "                                  'EC:1.8.1.4',\n",
    "                                  'citric acid cycle'),\n",
    "                                  \n",
    "                                  ('phosphoenolpyruvate carboxykinase (ATP)',\n",
    "                                  'converts oxaloacetate into phosphoenolpyruvate and carbon dioxide using ATP',\n",
    "                                  'EC:4.1.1.49',\n",
    "                                  'citric acid cycle'),\n",
    "                                  \n",
    "                                  ('phosphoenolpyruvate carboxykinase (GTP)',\n",
    "                                  'converts oxaloacetate into phosphoenolpyruvate and carbon dioxide using GTP',\n",
    "                                  '4.1.1.32',\n",
    "                                  'citric acid cycle');\"\"\")\n",
    "conn.commit()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Making Pathways table\n",
    "c.execute(\"\"\"DROP TABLE PATHWAYS\"\"\")\n",
    "c.execute(\"\"\"CREATE TABLE PATHWAYS (name TEXT,\n",
    "                                   description TEXT);\"\"\")\n",
    "conn.commit()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Insert data into PATHWAYS table\n",
    "c.execute(\"\"\"INSERT INTO PATHWAYS (name,\n",
    "                                  description)\n",
    "                          VALUES ('glycolysis',\n",
    "                                  'convert C6H12O6 into pyruvate'),\n",
    "                                  ('the citirc acid cycle',\n",
    "                                   'released stored energy through oxidation of acetyl CoA'),\n",
    "                                  ('pentose phosphate pathway',\n",
    "                                  'oxidation of glucose-6P and generates a ribulose-5P');\"\"\")\n",
    "\n",
    "conn.commit()\n",
    "                                  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Fetching data from the table\n",
    "\n",
    "In order to check what data we had input into each of the tables, SELECT command is used to select the desired table, fetchall is used to fetch all the data from that table selected. Below are data from the three tables recorded in the database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "c.execute('SELECT * FROM GENES')\n",
    "print (c.fetchall())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('phosphoglucomutase', 'transfers a phosphate group on an -D-glucose monomer from the 1 to the 6 position', 'EC:5.4.2.2', 'glycolysis'), ('Aldose 1-epimerase', 'catalyzes the chemical reaction between alpha-D-glucose and beta-D-glucose', 'EC:5.1.3.3', 'glycolysis'), ('Enolase', 'catalyze the conversion of 2-phosphoglycerate(2-PG) to phosphoenolpyruvate(PEP)', 'EC:4.2.1.11', 'glycolysis'), ('Glucose-6-phosphate isomerase', 'interconvert glucose 6-phosphate and fructose 6-phosphate', 'EC:5.3.1.9', 'glycolysis'), ('Glucose-6-phosphate isomerase', 'interconvert glucose 6-phosphate and fructose 6-phosphate', 'EC:5.3.1.9', 'pentose phosphate pathway'), ('6-phosphofructokinase', 'convert fructose-6-phosphate into fructose 1,6-bisphosphate', 'EC:2.7.1.11', 'pentose phosphate pathway'), ('Fructose-bisphosphate aldolase', 'splits the aldol and fructose 1,6-bisphosphate into the triose phosphates dihydroxyacetone phosphate(DHAP) and glyceraldehyde 3-phosphate(G3P)', 'EC:4.1.2.13', 'pentose phosphate pathway'), ('Fructose-1,6-bisphosphatase', 'converts fructose-1,6-bisphosphate to fructose 6-phosphate', 'EC:3.1.3.11', 'pentose phosphate pathway'), ('pyruvate dehydrogenase E1 component', 'pyruvate decarboxylation and reductive acetylation of lipoic acid', 'EC:1.2.4.1', 'citric acid cycle'), ('pyruvate dehydrogenase E2 component', 'bind on lipoate-thioester and by transacylation produce acetyl-CoA', 'EC:2.3.1.12', 'citric acid cycle'), ('dihydrolipoamide dehydrogenase', 'oxidize dihydrolipoamide to lipoamide', 'EC:1.8.1.4', 'citric acid cycle'), ('phosphoenolpyruvate carboxykinase (ATP)', 'converts oxaloacetate into phosphoenolpyruvate and carbon dioxide using ATP', 'EC:4.1.1.49', 'citric acid cycle'), ('phosphoenolpyruvate carboxykinase (GTP)', 'converts oxaloacetate into phosphoenolpyruvate and carbon dioxide using GTP', '4.1.1.32', 'citric acid cycle')]\n"
     ]
    }
   ],
   "source": [
    "c.execute('SELECT * FROM ENZYMES')\n",
    "print (c.fetchall())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('glycolysis', 'convert C6H12O6 into pyruvate'), ('the citirc acid cycle', 'released stored energy through oxidation of acetyl CoA'), ('pentose phosphate pathway', 'oxidation of glucose-6P and generates a ribulose-5P')]\n"
     ]
    }
   ],
   "source": [
    "c.execute('SELECT * FROM PATHWAYS')\n",
    "print (c.fetchall())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
