{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Importing tools \n",
    "SQL commands are used throughout this lab in order to create, select, drop or insert data into the tables made (Tables: GENES, ENZYMES, PATHWAYS). Sqlite3 is imported in order to introduce the SQL commands. \"Database\" is made referring to the database made to contain the tables. The cursor is for reading and performing actions for the database made. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sqlite3\n",
    "conn= sqlite3.connect('Database.db')\n",
    "c = conn.cursor()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Search for data imported from Entrez"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Finding name of enzymes in each pathway\n",
    "Finding the enzymes for each pathway by examining the KEGG Glycolysis / Gluconeogenesis - Reference pathway and picking out several chosen enzyme commission numbers indicated for each pathway. Clicked in those EC numbers to obtain the name of the enzymes.\n",
    "\n",
    "Googled for each enzyme's function. \n",
    "\n",
    "Below is a sample test to find out information about an enzyme using Entrez. A handle is created to first use esearch to find out the target enzyme in a certain organism. Then doing a for loop to fetch and print out the information wanted for the enzyme\n",
    "\n",
    "Noticing that the db.source gives us a reference of the gene accession number that is used to encode the enzyme."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: XP_023167159.1\n",
      "Name: XP_023167159\n",
      "Description: phosphoglucomutase [Drosophila hydei]\n",
      "Database cross-references: BioProject:PRJNA422293\n",
      "Number of features: 8\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=18-DEC-2017\n",
      "/accessions=['XP_023167159']\n",
      "/sequence_version=1\n",
      "/db_source=REFSEQ: accession XM_023311391.1\n",
      "/keywords=['RefSeq']\n",
      "/source=Drosophila hydei\n",
      "/organism=Drosophila hydei\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila']\n",
      "/comment=MODEL REFSEQ:  This record is predicted by automated computational\n",
      "analysis. This record is derived from a genomic sequence\n",
      "(NW_019378783.1) annotated using gene prediction method: Gnomon.\n",
      "Also see:\n",
      "    Documentation of NCBI's Annotation Process\n",
      "                               100\n",
      "                               pipeline\n",
      "COMPLETENESS: full length.\n",
      "/structured_comment=OrderedDict([('Genome-Annotation-Data', OrderedDict([('Annotation Provider', 'NCBI'), ('Annotation Status', 'Full annotation'), ('Annotation Name', 'Drosophila hydei Annotation Release'), ('Annotation Version', '100'), ('Annotation Pipeline', 'NCBI eukaryotic genome annotation'), ('Annotation Software Version', '8.0'), ('Annotation Method', 'Best-placed RefSeq; Gnomon'), ('Features Annotated', 'Gene; mRNA; CDS; ncRNA')]))])\n",
      "Seq('MSLSVETKLTTTYEGQKPGTSGLRKKVKVFTQPNYTENFVQCILDANGAALVGS...VIT', IUPACProtein())\n",
      "ID: XP_017955234.1\n",
      "Name: XP_017955234\n",
      "Description: PREDICTED: phosphoglucomutase [Drosophila navojoa]\n",
      "Database cross-references: BioProject:PRJNA340011\n",
      "Number of features: 3\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-SEP-2016\n",
      "/accessions=['XP_017955234']\n",
      "/sequence_version=1\n",
      "/db_source=REFSEQ: accession XM_018099745.1\n",
      "/keywords=['RefSeq']\n",
      "/source=Drosophila navojoa\n",
      "/organism=Drosophila navojoa\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila']\n",
      "/comment=MODEL REFSEQ:  This record is predicted by automated computational\n",
      "analysis. This record is derived from a genomic sequence\n",
      "(NW_017181430.1) annotated using gene prediction method: Gnomon.\n",
      "Also see:\n",
      "    Documentation of NCBI's Annotation Process\n",
      "                               Release 100\n",
      "                               pipeline\n",
      "COMPLETENESS: full length.\n",
      "/structured_comment=OrderedDict([('Genome-Annotation-Data', OrderedDict([('Annotation Provider', 'NCBI'), ('Annotation Status', 'Full annotation'), ('Annotation Version', 'Drosophila navojoa Annotation'), ('Annotation Pipeline', 'NCBI eukaryotic genome annotation'), ('Annotation Software Version', '7.1'), ('Annotation Method', 'Best-placed RefSeq; Gnomon'), ('Features Annotated', 'Gene; mRNA; CDS; ncRNA')]))])\n",
      "Seq('MEEIFDFGKLHDFVSGKSSGKPLKMRIDSMNGVTGAYVREIFLNRLGATEASVV...VIT', IUPACProtein())\n",
      "ID: XP_017863552.1\n",
      "Name: XP_017863552\n",
      "Description: PREDICTED: phosphoglucomutase [Drosophila arizonae]\n",
      "Database cross-references: BioProject:PRJNA339865\n",
      "Number of features: 3\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=29-AUG-2016\n",
      "/accessions=['XP_017863552']\n",
      "/sequence_version=1\n",
      "/db_source=REFSEQ: accession XM_018008063.1\n",
      "/keywords=['RefSeq']\n",
      "/source=Drosophila arizonae\n",
      "/organism=Drosophila arizonae\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila']\n",
      "/comment=MODEL REFSEQ:  This record is predicted by automated computational\n",
      "analysis. This record is derived from a genomic sequence\n",
      "(NW_017127684.1) annotated using gene prediction method: Gnomon.\n",
      "Also see:\n",
      "    Documentation of NCBI's Annotation Process\n",
      "                               Release 100\n",
      "                               pipeline\n",
      "COMPLETENESS: full length.\n",
      "/structured_comment=OrderedDict([('Genome-Annotation-Data', OrderedDict([('Annotation Provider', 'NCBI'), ('Annotation Status', 'Full annotation'), ('Annotation Version', 'Drosophila arizonae Annotation'), ('Annotation Pipeline', 'NCBI eukaryotic genome annotation'), ('Annotation Software Version', '7.1'), ('Annotation Method', 'Best-placed RefSeq; Gnomon'), ('Features Annotated', 'Gene; mRNA; CDS; ncRNA')]))])\n",
      "Seq('MSLNVETKQTTPYEGQKPGTSGLRKKVKVFTQPNYTENFVQCILEANGTALVGS...VIT', IUPACProtein())\n",
      "ID: AAL08568.1\n",
      "Name: AF416984_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=30-SEP-2001\n",
      "/accessions=['AAL08568']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF416984.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Doxycycline-induced expression of sense and inverted-repeat constructs modulates phosphoglucomutase gene expression in adult Drosophila melanogaster with effects on life span', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAL08567.1\n",
      "Name: AF416983_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=30-SEP-2001\n",
      "/accessions=['AAL08567']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF416983.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Doxycycline-induced expression of sense and inverted-repeat constructs modulates phosphoglucomutase gene expression in adult Drosophila melanogaster with effects on life span', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVGIVTTKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAL08566.1\n",
      "Name: AF416982_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=30-SEP-2001\n",
      "/accessions=['AAL08566']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF416982.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Doxycycline-induced expression of sense and inverted-repeat constructs modulates phosphoglucomutase gene expression in adult Drosophila melanogaster with effects on life span', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAL08565.1\n",
      "Name: AF416981_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=30-SEP-2001\n",
      "/accessions=['AAL08565']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF416981.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Doxycycline-induced expression of sense and inverted-repeat constructs modulates phosphoglucomutase gene expression in adult Drosophila melanogaster with effects on life span', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: XP_002062067.1\n",
      "Name: XP_002062067\n",
      "Description: phosphoglucomutase [Drosophila willistoni]\n",
      "Database cross-references: BioProject:PRJNA29997\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=04-DEC-2017\n",
      "/accessions=['XP_002062067']\n",
      "/sequence_version=1\n",
      "/db_source=REFSEQ: accession XM_002062031.3\n",
      "/keywords=['RefSeq']\n",
      "/source=Drosophila willistoni\n",
      "/organism=Drosophila willistoni\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Endopterygota', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/comment=MODEL REFSEQ:  This record is predicted by automated computational\n",
      "analysis. This record is derived from a genomic sequence\n",
      "(NW_002032468.1) annotated using gene prediction method: Gnomon,\n",
      "supported by EST evidence.\n",
      "Also see:\n",
      "    Documentation of NCBI's Annotation Process\n",
      "                               Release 101\n",
      "                               pipeline\n",
      "COMPLETENESS: full length.\n",
      "/structured_comment=OrderedDict([('Genome-Annotation-Data', OrderedDict([('Annotation Provider', 'NCBI'), ('Annotation Status', 'Full annotation'), ('Annotation Name', 'Drosophila willistoni Annotation'), ('Annotation Version', '101'), ('Annotation Pipeline', 'NCBI eukaryotic genome annotation'), ('Annotation Software Version', '8.0'), ('Annotation Method', 'Best-placed RefSeq; Gnomon'), ('Features Annotated', 'Gene; mRNA; CDS; ncRNA')]))])\n",
      "Seq('MSLSVETVQTKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQCILDANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAG44943.1\n",
      "Name: AF290356_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44943']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290356.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAG44942.1\n",
      "Name: AF290355_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44942']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290355.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAG44941.1\n",
      "Name: AF290354_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44941']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290354.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAG44940.1\n",
      "Name: AF290353_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44940']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290353.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAG44939.1\n",
      "Name: AF290352_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44939']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290352.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAG44938.1\n",
      "Name: AF290351_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44938']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290351.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAG44937.1\n",
      "Name: AF290350_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44937']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290350.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAG44936.1\n",
      "Name: AF290349_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44936']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290349.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: AAG44935.1\n",
      "Name: AF290348_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44935']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290348.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALIGS...VIT', IUPACProtein())\n",
      "ID: AAG44934.1\n",
      "Name: AF290347_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44934']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290347.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAVLAGS...VIT', IUPACProtein())\n",
      "ID: AAG44933.1\n",
      "Name: AF290346_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44933']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290346.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n",
      "ID: AAG44932.1\n",
      "Name: AF290345_1\n",
      "Description: phosphoglucomutase [Drosophila melanogaster]\n",
      "Number of features: 9\n",
      "/topology=linear\n",
      "/data_file_division=INV\n",
      "/date=08-FEB-2001\n",
      "/accessions=['AAG44932']\n",
      "/sequence_version=1\n",
      "/db_source=accession AF290345.1\n",
      "/keywords=['']\n",
      "/source=Drosophila melanogaster (fruit fly)\n",
      "/organism=Drosophila melanogaster\n",
      "/taxonomy=['Eukaryota', 'Metazoa', 'Ecdysozoa', 'Arthropoda', 'Hexapoda', 'Insecta', 'Pterygota', 'Neoptera', 'Holometabola', 'Diptera', 'Brachycera', 'Muscomorpha', 'Ephydroidea', 'Drosophilidae', 'Drosophila', 'Sophophora']\n",
      "/references=[Reference(title='Extensive amino acid polymorphism at the pgm locus is consistent with adaptive protein evolution in Drosophila melanogaster', ...), Reference(title='Direct Submission', ...)]\n",
      "/comment=Method: conceptual translation supplied by author.\n",
      "Seq('MSLTVEIVATKPYEGQKPGTSGLRKKVKVFTQPNYTENFVQAILEANGAALAGS...VIT', IUPACProtein())\n"
     ]
    }
   ],
   "source": [
    "# This is a sample test to find out information about an enzyme\n",
    "from Bio import SeqIO\n",
    "from Bio import Entrez\n",
    "Entrez.email = 'joanne91218@berkeley.edu'\n",
    "handle = Entrez.esearch(db = 'protein',\n",
    "                       term = 'drosophila[ORGN] phosphoglucomutase',\n",
    "                       sort = 'relevance',\n",
    "                       idtype = 'acc')\n",
    "for i in Entrez.read(handle)['IdList']:\n",
    "    handle = Entrez.efetch(db = 'protein', id=i, rettype = 'gb', retmode = 'text')\n",
    "    temp = SeqIO.read(handle,'gb')\n",
    "    print(temp) \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Fetching data for genes\n",
    "After knowing all the enzyme names from above, I tried to search for each gene accession number for three organisms for each enzyme.\n",
    "\n",
    "By first making the lists for enzymes in each pathway,a function is defined to search on Entrez for each terms. Take in if statement to filter out the ones that the Entrez did come out with some results. If there is a result searched, then make a handle to fetch the informations that I want (db_source, which is the accession number in this case). Genes.append is use to put those accessions numbers found into the genes list that is defined.\n",
    "\n",
    "After running the function for each enzyme in each pathway, define a list that will just give out the accession number (without the word: accession number). This is done by first splitting up the terms in the previous list, and obtaining purely the accession number and putting them into a new list. The new generated list is then printed. \n",
    "\n",
    "Note that there will be multiple genes corresponding to one enzyme.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "glycolysis_terms = ['homo sapiens[ORGN] phosphoglucomutase','drosophila[ORGN] phosphoglucomutase','Escherichia coli[ORGN] phosphoglucomutase',\n",
    "                  'homo sapiens[ORGN] aldose 1-epimerase','drosophila[ORGN] aldose 1-epimerase','Escherichia coli[ORGN] aldose 1-epimerase',\n",
    "                  'homo sapiens[ORGN] enolase','drosophila[ORGN] enolase','Escherichia coli[ORGN] enolase',\n",
    "                  'homo sapiens[ORGN] glucose-6-phosphate isomerase', 'drosophila[ORGN] glucose-6-phosphate isomerase','Escherichia coli[ORGN] glucose-6-phosphate isomerase']\n",
    "pentose_terms = ['homo sapiens[ORGN] glucose-6-phosphate isomerase','drosophila[ORGN] glucose-6-phosphate isomerase','Escherichia coli[ORGN] glucose-6-phosphate isomerase',\n",
    "                 'homo sapiens[ORGN] 6-phosphofructokinase', 'drosophila[ORGN] 6-phosphofructokinase', 'Escherichia coli[ORGN] 6-phosphofructokinase',\n",
    "                 'homo sapiens[ORGN] fructose-bisphosphate aldolase','drosophila[ORGN] fructose-bisphosphate aldolase', 'Escherichia coli[ORGN] fructose-bisphosphate aldolase',\n",
    "                 'homo sapiens[ORGN] fructose-1,6-bisphosphatase','drosophila[ORGN] fructose-1,6-bisphosphatase', 'Escherichia coli[ORGN] fructose-1,6-bisphosphatase']\n",
    "citric_terms = ['homo sapiens[ORGN] pyruvate dehydrogenase E1','drosophila[ORGN] pyruvate dehydrogenase E1','Escherichia coli[ORGN] pyruvate dehydrogenase E1'\n",
    "                'homo sapiens[ORGN] pyruvate dehydrogenase E2', 'drosophila[ORGN] pyruvate dehydrogenase E2', 'Escherichia coli[ORGN] pyruvate dehydrogenase E2',\n",
    "                'homo sapiens[ORGN] dihydrolipoamide dehydrogenase', 'drosophila[ORGN] dihydrolipoamide dehydrogenase', 'Escherichia coli[ORGN] dihydrolipoamide dehydrogenase',\n",
    "                'homo sapiens[ORGN] phosphoenolpyruvate carboxykinase', 'drosophila[ORGN] phosphoenolpyruvate carboxykinase', 'Escherichia coli[ORGN] phosphoenolpyruvate carboxykinase']\n",
    "# define three lists corresponding to enzymes for each organism in each pathway\n",
    "\n",
    "genes=[]       #define an empty list\n",
    "from Bio import SeqIO\n",
    "from Bio import Entrez\n",
    "Entrez.email = 'joanne91218@berkeley.edu'\n",
    "\n",
    "def find_gene_accession(cycle_name, cycle_list):\n",
    "    for t in cycle_list:\n",
    "        handle = Entrez.esearch(db='protein',\n",
    "                        term=t,\n",
    "                        sort='relevance',\n",
    "                        idtype='acc')\n",
    "        results = Entrez.read(handle)['IdList']\n",
    "        if results:\n",
    "            first_result = results[0]\n",
    "            handle=Entrez.efetch(db='protein', id=first_result, rettype='gb', retmode='text')\n",
    "            temp = SeqIO.read(handle, 'gb')\n",
    "            gene_acc = temp.annotations['db_source']     #extract db_source from the annotations\n",
    "            genes.append(gene_acc)\n",
    "\n",
    "# do the functions for each pathway\n",
    "find_gene_accession(\"Glycolysis\",glycolysis_terms)\n",
    "find_gene_accession(\"Pentose phosphate pathway\",pentose_terms)\n",
    "find_gene_accession(\"Citric acid cycle\", citric_terms)\n",
    "\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['FUIG01000013.1', 'XM_023311391.1', 'AFAS01000018.1', 'NM_138801.2', 'XM_023317388.1', 'CP026473.1', 'X66610.1', 'XM_023310563.1', 'X82400.1', 'PROSITE:PS51463', 'XM_023313082.1', 'PROSITE:PS51463', 'PROSITE:PS51463', 'XM_023313082.1', 'PROSITE:PS51463', 'FUIG01000013.1', 'NM_001273918.1', 'AFAV01000079.1', 'X07292.1', 'XM_023316276.1', 'AKNI01000030.1', 'L10320.1', 'LC058525.1', 'ADWR01000051.1', 'NM_005390.4', 'XM_023318640.1', 'XM_015178001.1', 'PROSITE:PS51826', 'AH003583.2', 'FJ907946.1', 'CP002516.1', 'L12760.1', 'NM_079060.3', 'AXTZ01000072.1']\n"
     ]
    }
   ],
   "source": [
    "gene_acc = []     # define an empty list to contain the gene accession number\n",
    "\n",
    "for g in genes:\n",
    "    length = g.split()\n",
    "    gene_acc.append(length[-1])      #eliminate the word accession number, keeping purely the numbers we want\n",
    "    \n",
    "print(gene_acc)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Creating tables and inserting data for genes, enzymes and pathways\n",
    "\n",
    "Gene Table:\tStart with name, description, organism, and nucleotide sequence. Additional fields might include chromosome, start and end position, strand, and translated sequence.\n",
    "\n",
    "Enzyme\tTable:\tIncludes name, function, and enzyme\tcommission (EC)\tnumber, and pathways the enzymes belong to.\t\n",
    "\n",
    "Pathway\tTable: name and description.\t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Making GENES table\n",
    "c.execute(\"\"\"DROP TABLE GENES\"\"\") # to prevent the table repeatedly create or insert data after running execution each time, a drop command is used\n",
    "c.execute(\"\"\"CREATE TABLE GENES (name TEXT,\n",
    "                                description TEXT,\n",
    "                                organism TEXT,\n",
    "                                nucleotide sequences TEXT);\"\"\")      # creating a table named GENES\n",
    "conn.commit()   #commit changes to the database"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Inserting data into GENES table\n",
    "From the gene accession number lists I obtained above, define a new function to search in Entrez for the detailed genes information for each organism for each enzyme.Take in if statement to filter out the ones that the Entrez did come out with some results. If there is a result searched, then make a handle to fetch the informations that I want.\n",
    "\n",
    "Finally, execute with SQL command to insert data correspondingly into the GENES table. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fill_table(gene_accnum):\n",
    "    for t in gene_accnum:\n",
    "        handle = Entrez.esearch(db='nucleotide',\n",
    "                        term=t,\n",
    "                        sort='relevance',\n",
    "                        idtype='acc')\n",
    "        results = Entrez.read(handle)['IdList']\n",
    "        if results:\n",
    "            first_result = results[0]\n",
    "            handle=Entrez.efetch(db='nucleotide', id=first_result, rettype='gb', retmode='text')\n",
    "            temp = SeqIO.read(handle, 'gb')\n",
    "            c.execute(\"INSERT INTO GENES VALUES (?, ?, ?, ?)\", \n",
    "                      (temp.name, temp.description, temp.annotations['organism'], str(temp.seq)))\n",
    "        else:\n",
    "            continue\n",
    "\n",
    "\n",
    "fill_table(gene_acc)\n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Making ENZYMES table\n",
    "c.execute(\"\"\"DROP TABLE ENZYMES\"\"\")\n",
    "c.execute(\"\"\"CREATE TABLE ENZYMES (name TEXT,\n",
    "                                 function TEXT,\n",
    "                                 Enzyme commission number TEXT,\n",
    "                                 Pathway TEXT);\"\"\")\n",
    "conn.commit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Inserting data into ENZYMES table\n",
    "c.execute(\"\"\"INSERT INTO ENZYMES  VALUES('phosphoglucomutase',\n",
    "                                  'transfers a phosphate group on an α-D-glucose monomer from the 1 to the 6 position',\n",
    "                                  'EC:5.4.2.2',\n",
    "                                  'glycolysis'),\n",
    "                                  \n",
    "                                  ('Aldose 1-epimerase',\n",
    "                                  'catalyzes the chemical reaction between alpha-D-glucose and beta-D-glucose',\n",
    "                                  'EC:5.1.3.3',\n",
    "                                  'glycolysis'),\n",
    "                                  \n",
    "                                  ('Enolase',\n",
    "                                  'catalyze the conversion of 2-phosphoglycerate(2-PG) to phosphoenolpyruvate(PEP)',\n",
    "                                  'EC:4.2.1.11',\n",
    "                                  'glycolysis'),\n",
    "                                  \n",
    "                                  ('Glucose-6-phosphate isomerase',\n",
    "                                  'interconvert glucose 6-phosphate and fructose 6-phosphate',\n",
    "                                  'EC:5.3.1.9',\n",
    "                                  'glycolysis'),\n",
    "                                  \n",
    "                                  ('Glucose-6-phosphate isomerase',\n",
    "                                  'interconvert glucose 6-phosphate and fructose 6-phosphate',\n",
    "                                  'EC:5.3.1.9',\n",
    "                                  'pentose phosphate pathway'),\n",
    "                                  \n",
    "                                  ('6-phosphofructokinase',\n",
    "                                  'convert fructose-6-phosphate into fructose 1,6-bisphosphate',\n",
    "                                  'EC:2.7.1.11',\n",
    "                                  'pentose phosphate pathway'),\n",
    "                                  \n",
    "                                  ('Fructose-bisphosphate aldolase',\n",
    "                                  'splits the aldol and fructose 1,6-bisphosphate into the triose phosphates dihydroxyacetone phosphate(DHAP) and glyceraldehyde 3-phosphate(G3P)',\n",
    "                                  'EC:4.1.2.13',\n",
    "                                  'pentose phosphate pathway'),\n",
    "                                  \n",
    "                                  ('Fructose-1,6-bisphosphatase',\n",
    "                                  'converts fructose-1,6-bisphosphate to fructose 6-phosphate',\n",
    "                                  'EC:3.1.3.11',\n",
    "                                  'pentose phosphate pathway'),\n",
    "                                  \n",
    "                                  ('pyruvate dehydrogenase E1 component',\n",
    "                                  'pyruvate decarboxylation and reductive acetylation of lipoic acid',\n",
    "                                  'EC:1.2.4.1',\n",
    "                                  'citric acid cycle'),\n",
    "                                  \n",
    "                                  ('pyruvate dehydrogenase E2 component',\n",
    "                                  'bind on lipoate-thioester and by transacylation produce acetyl-CoA',\n",
    "                                  'EC:2.3.1.12',\n",
    "                                  'citric acid cycle'),\n",
    "                                  \n",
    "                                  ('dihydrolipoamide dehydrogenase',\n",
    "                                  'oxidize dihydrolipoamide to lipoamide',\n",
    "                                  'EC:1.8.1.4',\n",
    "                                  'citric acid cycle'),\n",
    "                                  \n",
    "                                  ('phosphoenolpyruvate carboxykinase (ATP)',\n",
    "                                  'converts oxaloacetate into phosphoenolpyruvate and carbon dioxide using ATP',\n",
    "                                  'EC:4.1.1.49',\n",
    "                                  'citric acid cycle'),\n",
    "                                  \n",
    "                                  ('phosphoenolpyruvate carboxykinase (GTP)',\n",
    "                                  'converts oxaloacetate into phosphoenolpyruvate and carbon dioxide using GTP',\n",
    "                                  '4.1.1.32',\n",
    "                                  'citric acid cycle');\"\"\")\n",
    "conn.commit()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Making Pathways table\n",
    "c.execute(\"\"\"DROP TABLE PATHWAYS\"\"\")\n",
    "c.execute(\"\"\"CREATE TABLE PATHWAYS (name TEXT,\n",
    "                                   description TEXT);\"\"\")\n",
    "conn.commit()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Insert data into PATHWAYS table\n",
    "c.execute(\"\"\"INSERT INTO PATHWAYS (name,\n",
    "                                  description)\n",
    "                          VALUES ('glycolysis',\n",
    "                                  'convert C6H12O6 into pyruvate'),\n",
    "                                  ('the citirc acid cycle',\n",
    "                                   'released stored energy through oxidation of acetyl CoA'),\n",
    "                                  ('pentose phosphate pathway',\n",
    "                                  'oxidation of glucose-6P and generates a ribulose-5P');\"\"\")\n",
    "\n",
    "conn.commit()\n",
    "                                  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Fetching data from the table\n",
    "\n",
    "In order to check what data we had input into each of the tables, SELECT command is used to select the desired table, fetchall is used to fetch all the data from that table selected. Below are data from the three tables recorded in the database."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To print out gene table:\n",
    "\n",
    "c.execute('SELECT * FROM GENES')\n",
    "\n",
    "print (c.fetchall())\n",
    "\n",
    "supposely should be using this command, however the gene table is too large for it to print out everything from the gene table."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('phosphoglucomutase', 'transfers a phosphate group on an α-D-glucose monomer from the 1 to the 6 position', 'EC:5.4.2.2', 'glycolysis'), ('Aldose 1-epimerase', 'catalyzes the chemical reaction between alpha-D-glucose and beta-D-glucose', 'EC:5.1.3.3', 'glycolysis'), ('Enolase', 'catalyze the conversion of 2-phosphoglycerate(2-PG) to phosphoenolpyruvate(PEP)', 'EC:4.2.1.11', 'glycolysis'), ('Glucose-6-phosphate isomerase', 'interconvert glucose 6-phosphate and fructose 6-phosphate', 'EC:5.3.1.9', 'glycolysis'), ('Glucose-6-phosphate isomerase', 'interconvert glucose 6-phosphate and fructose 6-phosphate', 'EC:5.3.1.9', 'pentose phosphate pathway'), ('6-phosphofructokinase', 'convert fructose-6-phosphate into fructose 1,6-bisphosphate', 'EC:2.7.1.11', 'pentose phosphate pathway'), ('Fructose-bisphosphate aldolase', 'splits the aldol and fructose 1,6-bisphosphate into the triose phosphates dihydroxyacetone phosphate(DHAP) and glyceraldehyde 3-phosphate(G3P)', 'EC:4.1.2.13', 'pentose phosphate pathway'), ('Fructose-1,6-bisphosphatase', 'converts fructose-1,6-bisphosphate to fructose 6-phosphate', 'EC:3.1.3.11', 'pentose phosphate pathway'), ('pyruvate dehydrogenase E1 component', 'pyruvate decarboxylation and reductive acetylation of lipoic acid', 'EC:1.2.4.1', 'citric acid cycle'), ('pyruvate dehydrogenase E2 component', 'bind on lipoate-thioester and by transacylation produce acetyl-CoA', 'EC:2.3.1.12', 'citric acid cycle'), ('dihydrolipoamide dehydrogenase', 'oxidize dihydrolipoamide to lipoamide', 'EC:1.8.1.4', 'citric acid cycle'), ('phosphoenolpyruvate carboxykinase (ATP)', 'converts oxaloacetate into phosphoenolpyruvate and carbon dioxide using ATP', 'EC:4.1.1.49', 'citric acid cycle'), ('phosphoenolpyruvate carboxykinase (GTP)', 'converts oxaloacetate into phosphoenolpyruvate and carbon dioxide using GTP', '4.1.1.32', 'citric acid cycle')]\n"
     ]
    }
   ],
   "source": [
    "#print out enzyme table\n",
    "c.execute('SELECT * FROM ENZYMES')\n",
    "print (c.fetchall())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('glycolysis', 'convert C6H12O6 into pyruvate'), ('the citirc acid cycle', 'released stored energy through oxidation of acetyl CoA'), ('pentose phosphate pathway', 'oxidation of glucose-6P and generates a ribulose-5P')]\n"
     ]
    }
   ],
   "source": [
    "# print out pathway table\n",
    "c.execute('SELECT * FROM PATHWAYS')\n",
    "print (c.fetchall())\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
